<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="Content-Style-Type" content="text/css" />
<meta name="generator" content="pandoc" />



<title>1: Introduction</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="libs/bootstrap-2.3.2/css/readable.min.css" rel="stylesheet" />
<link href="libs/bootstrap-2.3.2/css/bootstrap-responsive.min.css" rel="stylesheet" />
<script src="libs/bootstrap-2.3.2/js/bootstrap.min.js"></script>
<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

/* offset scroll position for anchor links (for fixed navbar)  */
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="brand" href="index.html"><b>Population Genetics in R</b></a>
      <div class="nav-collapse collapse">
        <ul class="nav">
          <li class="dropdown">
            <a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>
            <ul class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="C01_Introduction.html">1. Introduction</a></li>
               <li><a href="C02_Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="C03_Data_Preparation.html">3. Data Preparation</a></li>
               <li><a href="C04_First_Steps.html">4. First steps</a></li>
               <li><a href="C05_Population_Hierarchies.html">5. Population hierarchies</a></li>
               <li><a href="C06_Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="C07_Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="C08_Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="C09_Pop_Structure.html">9. Population structure</a></li>
               <li><a href="C10_AMOVA.html">10. AMOVA</a></li>
               <li><a href="C11_DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
               </ul>
            </li>
          <li class="dropdown">
            <a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>
            <ul class = "dropdown-menu">
               <li><a href="A1_Data_sets.html">A1. Data sets</a></li>
               <li><a href="A2_Function_Glossary.html">A2. Function glossary</a></li>
            </ul>
          </li>
          
        <ul class="nav">
          <li><a href="Authors.html">About</a></li>
          <li class="dropdown">

        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </div>
</div>

<div class="container">
<div class="row">
<div class="span10 offset1">

<div id="header">
<h1 class="title">1: Introduction</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#why-use-r">Why use R?</a></li>
<li><a href="#population-genetics">Population genetics</a></li>
<li><a href="#the-special-case-of-clonal-organisms">The special case of clonal organisms</a></li>
<li><a href="#marker-systems-ploidy-and-other-kinds-of-data">Marker systems, ploidy and other kinds of data</a></li>
<li><a href="#applications-of-population-genetic-analyses">Applications of population genetic analyses</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>This primer provides a concise introduction to conducting applied analyses of population genetic data in R, with a special emphasis on non-model populations including clonal or partially clonal organisms. It is not meant to be a textbook on population genetics. Quite to the contrary, we refer the reader to several textbooks on population genetics <span class="citation">(1–3)</span>. Likewise, this book will not replace books on theory and statistics of population genetics <span class="citation">(4)</span>. The reader is thus expected to have a basic understanding of population genetic theory and applications. Finally, this primer is focused on traditional population genetics based on allele frequencies, rather than more sophisticated coalescent approaches <span class="citation">(1, 5, 6)</span>, although some of the material covered here will apply. In a nutshell, this primer provides a valuable resource for tackling the nitty-gritty analysis of populations that do not necessarily conform to textbook genetics and might or might not be in Hardy-Weinberg equilibrium and panmixia.</p>
<p>This primer is geared towards biologists interested in analyzing their populations. This primer does not require extensive knowledge of programming in R, but the user is expected to install R and all packages required for this primer.</p>
<div id="why-use-r" class="section level2">
<h2>Why use R?</h2>
<p>Until recently, one of the more tedious aspects of conducting a population genetic analysis was the need for repeated reformatting data to conduct different, complimentary analyses in different programs. Often, these programs only ran on one platform. Now, R provides a toolbox with its packages that allows analysis of most data conveniently without tedious reformatting on all major computing platforms including Microsoft Windows, Linux, and Apple’s OS X. R is an open source statistical programming and graphing language that includes tools for statistical, population genetic, genomic, phylogenetic, and comparative genomic analyses. This primer is for those of us that want to conduct applied analyses of populations and make full use of the palette of tools available in R.</p>
<p>Note that the R user community is very active and that both R and its packages are regularly updated, critically modified, and noted as deprecated (no longer updated) as appropriate.</p>
<blockquote>
<p>Any R user needs to make sure all components are up-to-date and that versions are compatible.</p>
</blockquote>
</div>
<div id="population-genetics" class="section level2">
<h2>Population genetics</h2>
<p>Traditional population genetics is based on analysis of observed allele frequencies compared to frequencies expected, assuming a population genetic model. For example, under a <strong>Wright-Fisher model</strong> you might expect to see populations of diploid individuals that reproduce sexually, with non-overlapping generations. This model ignores effects such as mutation, recombination, selection or changes in population size or structure. More complex models can incorporate different aspects of effects observed in real populations. However, most of these models assume that populations reproduce sexually.</p>
<p>Here, we briefly review some terminology but assume that the reader has a basic understanding of theory and applications of population genetics.</p>
<p>A <strong>locus</strong> is a position in the genome where we can observe one or several alleles in different individuals. Loci used in population genetics are assumed to be selectively neutral and can be an anonymous or non-coding region such as a microsatellite locus (SSR), a single nucleotide polymorphism (SNP) or the presence/absence of a band on a gel. A <strong>genotype</strong> is the combination of alleles carried by a given individual at a particular set of loci. Individuals carrying the same set of alleles are considered to have the same <strong>multilocus genotype</strong> <span class="math">\(MLG\)</span>.</p>
<p>Basic metrics of interest in populations are polymorphisms, allele frequencies and genotype frequencies. Polymorphism can be estimated in several ways, such as the total number of loci observed that have more than one allele. The frequency of an allele is calculated as the number of allele copies observed in a population divided by the total number of individuals, times the ploidy (<span class="math">\(N\)</span> in haploids or <span class="math">\(2N\)</span> in diploids) in the population. For diploids we would observe frequency <span class="math">\(f_{A}\)</span> and <span class="math">\(f_{a}\)</span> for alleles <span class="math">\(A\)</span> and <span class="math">\(a\)</span>, respectively:</p>
<p><span class="math">\(f_{A} = \frac{N_{A}}{2N}\)</span> and <span class="math">\(f_{a} = \frac{N_{a}}{2N}\)</span></p>
<p>where <span class="math">\(N_{A}\)</span> are the numbers of allele <span class="math">\(A\)</span> segregating in the population genotyped.</p>
<p>By definition, at a biallelic locus frequencies sum to 1:</p>
<p><span class="math">\(f_{A} + f_{a} = 1\)</span>.</p>
<p>Genotype frequencies are the relative frequencies of each <span class="math">\(MLG\)</span> observed in a population. Thus, for diploids at a <strong>biallelic locus</strong> we can observe three genotypes: <span class="math">\(AA\)</span>, <span class="math">\(Aa\)</span>, and <span class="math">\(aa\)</span> and their respective frequencies:</p>
<p><span class="math">\(f_{AA} = \frac{N_{AA}}{2N}\)</span> and <span class="math">\(f_{Aa} = \frac{N_{Aa}}{2N}\)</span> and <span class="math">\(f_{aa} = \frac{N_{aa}}{2N}\)</span></p>
<p>These frequencies again add up to 1. In diploid organisms we can also calculate the frequency of <strong>homozygotes</strong> (<span class="math">\(AA\)</span> or <span class="math">\(aa\)</span>) or <strong>heterozygotes</strong> (<span class="math">\(Aa\)</span>) as the proportion of individuals falling into each class. The proportion of individuals that are heterozygous is given by <span class="math">\(f_{Aa}\)</span> while the homozygous proportion is given by <span class="math">\(1-f_{Aa} = f_{AA}+f_{aa}\)</span>.</p>
<p>An important aspect of population structure is the heterozygosity observed within subdivided populations <span class="math">\(H_{S}\)</span> and across total populations <span class="math">\(H_{T}\)</span>. If we assume presence of two populations in Hardy-Weinberg equilibrium then frequencies of Allele <span class="math">\(A\)</span> in populations 1 and 2 pooled over both populations would be:</p>
<p><span class="math">\(f_{A} = \frac{2N_{1}f_{A1} + 2N_{2}f_{A2}}{2N_{1} + 2N_{2}}\)</span></p>
<p>where each allele frequency is multiplied by population size <span class="math">\(N_{1}\)</span> and <span class="math">\(N_{2}\)</span> and divided by total number of alleles <span class="math">\(2N_{1} + 2N_{2}\)</span>. Remember, that in a diploid population there are 2 alleles per locus per individual and hence every frequency is multiplied by 2.</p>
<p>Genetic differentiation of two populations (e.g. subdivision of populations) occurs if we observe differences in polymorphism, allele frequencies, genotype frequencies, and heterozygosity. Wright’s <span class="math">\(F_{st}\)</span> is a commonly used measure of differences in heterozygosity observed in the overall population relative to the subdivided populations and is defined as the differences between <span class="math">\(H_{T}\)</span> and <span class="math">\(H_{S}\)</span> relative to <span class="math">\(H_{T}\)</span>:</p>
<p><span class="math">\(F_{ST} = \frac{H_{T}-H_{S}}{H_{T}}\)</span></p>
<p>If allele frequencies are identical in both subpopulations then <span class="math">\(H_{T} = H_{S)}\)</span>. If, on the other hand, allele frequencies vary between subdivided populations, then <span class="math">\(H_{T} &gt; H_{S)}\)</span> and populations are considered to be genetically differentiated.</p>
<p>Besides calculation of <span class="math">\(F_{st}\)</span> other approaches can be used to assess population structure including clustering. These approaches will be explored in other chapters.</p>
</div>
<div id="the-special-case-of-clonal-organisms" class="section level2">
<h2>The special case of clonal organisms</h2>
<p>Plants and microorganisms such as fungi, oomycetes, and bacteria often reproduce clonally or follow a mixed reproductive system, including various degrees of clonality and sexuality <span class="citation">(7)</span>. Traditional population genetic theory does not apply and these populations violate basic assumptions in the corresponding analyses. Thus, a different set of tools are required for analyses of clonal populations <span class="citation">(7–12)</span>. Typically these focus on metrics that are model free and do not assume random mating or random sampling of alleles. Model free metrics suitable for analyzing these populations include measures of genotypic diversity and evenness, and using genetic distance to cluster populations. Additional useful measures include indices of linkage disequiblibrium among markers that are used to infer if populations are reproducing sexually or clonally. We will explore these analyses bit by bit throughout this primer. The <em>poppr</em> R package was specifically developed to facilitate analyses of clonal populations <span class="citation">(13)</span>.</p>
</div>
<div id="marker-systems-ploidy-and-other-kinds-of-data" class="section level2">
<h2>Marker systems, ploidy and other kinds of data</h2>
<p>Population genetic data come in many shapes and forms and a good analysis needs to be tailored to the marker system, ploidy used and corresponding genetic models assumed <span class="citation">(11)</span>. Organisms can be haploid, diploid, with or without known phase, or polyploid. In the case of diploid species, marker systems can be dominant (AFLP, RFLP or RAPD data) or co-dominant (SSR, SNPs, allozymes) <span class="citation">(14)</span>. A locus can have two alleles (0/1 for AFLP; C/T for SNPs) or many alleles per locus (A/B/C…N for SSRs or allozymes). In each case the data has to be coded and analyzed differently. Mutation rates can also differ for different markers systems (SSR &gt;&gt; mitochondrial SNPs &gt; nuclear SNPs) <span class="citation">(11)</span>.</p>
</div>
<div id="applications-of-population-genetic-analyses" class="section level2">
<h2>Applications of population genetic analyses</h2>
<p>Population genetic analyses are tremendously valuable for answering questions ranging from determining best management practices to basic evolutionary questions <span class="citation">(11)</span>. For example, a typical concern when finding a new, invasive organism is whether it is introduced to the area or emerged from a resident population. Other questions of interest might include:</p>
<ul>
<li>Where is the center of origin?</li>
<li>Does this organism reproduce sexually?</li>
<li>Is there evidence of recombination?</li>
<li>Has the population gone through a genetic bottleneck?</li>
<li>Are populations structured by region, geography, micro-environment?</li>
<li>What are source and sink populations and what are the rates of migration?</li>
</ul>
<p>This primer will provide some of the basic tools needed to answer many of these questions for populations that are clonal or partially clonal. Note, however, that more powerful, complementary coalescent methods will not be covered here. We hope you enjoy following along.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!--- ref --->


<div class="references">
<p>1. Nielsen R, Slatkin M (2013) <em>An introduction to population genetics: Theory and applications</em> (Sinauer Associates, Incorporated) Available at: <a href="http://books.google.com/books?id=Iy08kgEACAAJ">http://books.google.com/books?id=Iy08kgEACAAJ</a>.</p>
<p>2. Templeton A (2006) <em>Population genetics and microevolutionary theory</em> (Wiley) Available at: <a href="http://books.google.com/books?id=tJVreQjInt0C">http://books.google.com/books?id=tJVreQjInt0C</a>.</p>
<p>3. Hartl D, Clark A (2007) <em>Principles of population genetics</em> (Sinauer Associates, Incorporated) Available at: <a href="http://books.google.com/books?id=SB1vQgAACAAJ">http://books.google.com/books?id=SB1vQgAACAAJ</a>.</p>
<p>4. Weir B (1996) <em>Genetic data analysis 2</em> (Sinauer Associates) Available at: <a href="http://books.google.com/books?id=e9QPAQAAMAAJ">http://books.google.com/books?id=e9QPAQAAMAAJ</a>.</p>
<p>5. Wakeley J (2009) <em>Coalescent theory: An introduction</em> (Roberts &amp; Company Publishers) Available at: <a href="http://books.google.com/books?id=x30RAgAACAAJ">http://books.google.com/books?id=x30RAgAACAAJ</a>.</p>
<p>6. Hein J, Schierup M, Wiuf C (2004) <em>Gene genealogies, variation and evolution: A primer in coalescent theory</em> (Oxford University Press, USA) Available at: <a href="http://books.google.com/books?id=QBC\_SFOamksC">http://books.google.com/books?id=QBC\_SFOamksC</a>.</p>
<p>7. Halkett F, Simon J-C, Balloux F (2005) Tackling the population genetics of clonal and partially clonal organisms. <em>Trends in Ecology &amp; Evolution</em> 20:194–201. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16701368">http://www.ncbi.nlm.nih.gov/pubmed/16701368</a>.</p>
<p>8. Anderson J, Kohn L (1995) Clonality in soilborne, plant-pathogenic fungi. <em>Annual Review of Phytopathology</em> 33:369–391. Available at: <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev.py.33.090195.002101">http://www.annualreviews.org/doi/abs/10.1146/annurev.py.33.090195.002101</a>.</p>
<p>9. Balloux F, Lehmann L, Mee<span>û</span>s T de (2003) The population genetics of clonal and partially clonal diploids. <em>Genetics</em> 164:1635–1644. Available at: <a href="http://www.genetics.org/content/164/4/1635">http://www.genetics.org/content/164/4/1635</a>.</p>
<p>10. De Mee<span>û</span>s T, Lehmann L, Balloux F (2006) Molecular epidemiology of clonal diploids: a quick overview and a short dIY (do it yourself) notice. <em>Infection, Genetics and Evolution</em> 6:163–170. Available at: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16290062">http://www.ncbi.nlm.nih.gov/pubmed/16290062</a>.</p>
<p>11. Grünwald N, Goss E (2011) Evolution and population genetics of exotic and re-emerging pathogens: Novel tools and approaches. <em>Annual Review of Phytopathology</em> 49:249–267. Available at: <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto">http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto</a>.</p>
<p>12. Milgroom M (1996) Recombination and the multilocus structure of fungal populations. <em>Annual review of phytopathology</em> 34:457–477. Available at: <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev.phyto.34.1.457">http://www.annualreviews.org/doi/abs/10.1146/annurev.phyto.34.1.457</a>.</p>
<p>13. Kamvar Z, Tabima J, Grünwald N (2014) <span class="math">\(Poppr\)</span>: an r package for genetic analysis of populations with clonal, partially clonal, and/or sexual reproduction. <em>PeerJ</em> 2:e281. Available at: <a href="http://dx.doi.org/10.7717/peerj.281">http://dx.doi.org/10.7717/peerj.281</a>.</p>
<p>14. Grünwald N, Goodwin S, Milgroom M, Fry W (2003) Analysis of genotypic diversity data for populations of microorganisms. <em>Phytopathology</em> 93:738–746. Available at: <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738</a>.</p>
</div>
</div>

</div>
</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with --self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "libs/mathjax-2.3.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
