<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>12: Discriminant analysis of principal components (DAPC)</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/readable.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Strata.html">5. Population strata and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="Minimum_Spanning_Networks.html">10. Minimum Spanning Networks</a></li>
               <li><a href="AMOVA.html">11. AMOVA</a></li>
               <li><a href="DAPC.html">12. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="Function_Glossary.html">A2. Function glossary</a></li>
            <li><a href="Intro_to_R.html">A3. Introduction to R</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">12: Discriminant analysis of principal components (DAPC)</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#dapc-analysis-of-the-h3n2-influenza-strains">DAPC analysis of the H3N2 influenza strains</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>Often we want to infer population structure by determining the number of clusters (groups) observed without prior knowledge. Several approaches can be used to infer groups such as for example K-means clustering, Bayesian clustering using STRUCTURE, and multivariate methods such as Discriminant Analysis of Principal Components (DAPC) <span class="citation">(Pritchard, Stephens &amp; Donnelly, 2000; Jombart, Devillard &amp; Balloux, 2010; Grünwald &amp; Goss, 2011)</span>. A STRUCTURE-like approach assumes that markers are not linked and that populations are panmictic <span class="citation">(Pritchard et al., 2000)</span>. To use model-free methods K-means clustering based on genetic distance or DAPC are more convenient approaches for populations that are clonal or partially clonal. Here we explore DAPC further.</p>
<div id="dapc-analysis-of-the-h3n2-influenza-strains" class="section level2">
<h2>DAPC analysis of the H3N2 influenza strains</h2>
<p>DAPC was pioneered by Jombart and colleagues <span class="citation">(Jombart et al., 2010)</span> and can be used to infer the number of clusters of genetically related individuals. In this multivariate statistical approach variance in the sample is partitioned into a between-group and within-group component, in an effort to maximize discrimination between groups. In DAPC, data is first transformed using a principal components analysis (PCA) and subsequently clusters are identified using discriminant analysis (DA). This tutorial is based on the <a href="http://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf">vignette</a> written by Thibaut Jombart. We encourage the user to explore this vignette further. The vignette can also be opened within R by executing <code>adegenetTutorial(&quot;dapc&quot;)</code>.</p>
<p>We will use the seasonal influenza dataset H3N2 data containing 1903 isolates genotyped for 125 SNPs located in the hemagglutinin segment. This dataset as well as the <code>dapc()</code> function is part of the <a href="http://adegenet.r-forge.r-project.org"><em>adegenet</em></a> package.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># DAPC requires the adegenet package. Let&#39;s load this package:</span>
<span class="kw">library</span>(<span class="st">&quot;adegenet&quot;</span>)
<span class="kw">data</span>(H3N2)      <span class="co"># load the H3N2 influenza data. Tye ?H3N2 for more info.</span>
<span class="kw">pop</span>(H3N2) &lt;-<span class="st"> </span><span class="kw">factor</span>(H3N2$other$epid)
dapc.H3N2 &lt;-<span class="st"> </span><span class="kw">dapc</span>(H3N2, <span class="dt">var.contrib=</span><span class="ot">TRUE</span>, <span class="dt">scale=</span><span class="ot">FALSE</span>, <span class="dt">n.pca=</span><span class="dv">30</span>, <span class="dt">n.da=</span><span class="dv">10</span>)
<span class="kw">scatter</span>(dapc.H3N2, <span class="dt">cell=</span><span class="dv">0</span>, <span class="dt">pch=</span><span class="dv">18</span>:<span class="dv">23</span>, <span class="dt">cstar=</span><span class="dv">0</span>, <span class="dt">mstree=</span><span class="ot">TRUE</span>, <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">lty=</span><span class="dv">2</span>)</code></pre>
<p><img src="DAPC_files/figure-html/unnamed-chunk-2-1.png" title="" alt="" width="960" /></p>
<p>The <code>dapc()</code> variables refer to:</p>
<ul>
<li>the dataset H3N2</li>
<li><code>var.contrib</code> when set to FALSE, flag removes contribution from original variables from output. We will look at this later to see which loci are responsible for separatig populations.</li>
<li><code>n.pca</code> is the number of axes retained in the Principal Component Analysis (PCA). If set to NULL selection is interactive, which we avoid here but can be explored by the user.</li>
<li><code>n.da</code> is the number of axes retained in the Discriminant Analysis (DA). If set to NULL selection is interactive, which we avoid here but can be explored by the user.</li>
</ul>
<p>The <code>scatter()</code> function is part of the <em>ade4</em> pacakge and plots results of a DAPC analysis.</p>
<p>As you can see, each year between 2001 to 2005 is a cluster of H3N2 strains separated by axis 1. In contrast, axis 2 separates the strains observed in the 2006 cluster from the clusters observed during 2001-5, indicating that the strains observed in 2006 are genetically distinct.</p>
<p>Next, let’s assess if there are alleles that most differentiate the 2006 cluster from those in other years.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">4</span>)
contrib &lt;-<span class="st"> </span><span class="kw">loadingplot</span>(dapc.H3N2$var.contr, <span class="dt">axis=</span><span class="dv">2</span>, <span class="dt">thres=</span>.<span class="dv">07</span>, <span class="dt">lab.jitter=</span><span class="dv">1</span>)</code></pre>
<p><img src="DAPC_files/figure-html/unnamed-chunk-3-1.png" title="" alt="" width="960" /></p>
<p>It looks like SNPs at position 384 and 906 are involved. Let’s check this further by looking at allele frequencies by year:</p>
<pre class="sourceCode r"><code class="sourceCode r">temp &lt;-<span class="st"> </span><span class="kw">seploc</span>(H3N2)
snp906 &lt;-<span class="st"> </span><span class="kw">tab</span>(temp[[<span class="st">&quot;906&quot;</span>]]) <span class="co"># tab{adegenet} returns a matrix of genotypes</span>
snp399 &lt;-<span class="st"> </span><span class="kw">tab</span>(temp[[<span class="st">&quot;399&quot;</span>]])
freq906 &lt;-<span class="st"> </span><span class="kw">apply</span>(snp906, <span class="dv">2</span>, function(e) <span class="kw">tapply</span>(e, <span class="kw">pop</span>(H3N2), mean, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))
freq399 &lt;-<span class="st"> </span><span class="kw">apply</span>(snp399, <span class="dv">2</span>, function(e) <span class="kw">tapply</span>(e, <span class="kw">pop</span>(H3N2), mean, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))
freq906</code></pre>
<pre><code>##            906.c     906.t
## 2001 0.000000000 1.0000000
## 2002 0.000000000 1.0000000
## 2003 0.000000000 1.0000000
## 2004 0.000000000 1.0000000
## 2005 0.002155172 0.9978448
## 2006 0.616071429 0.3839286</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">freq399</code></pre>
<pre><code>##            399.c     399.t
## 2001 0.000000000 1.0000000
## 2002 0.000000000 1.0000000
## 2003 0.000000000 1.0000000
## 2004 0.001848429 0.9981516
## 2005 0.002079002 0.9979210
## 2006 0.357142857 0.6428571</code></pre>
<p>Note that a new allele appeared in 2005 for SNP locus 906 and 2004 for locus 399 separating populations along axis 2.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="dt">mar=</span><span class="kw">c</span>(<span class="fl">5.1</span>,<span class="fl">4.1</span>,<span class="fl">4.1</span>,.<span class="dv">1</span>),<span class="dt">las=</span><span class="dv">3</span>)
<span class="kw">matplot</span>(freq906, <span class="dt">pch=</span><span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;t&quot;</span>), <span class="dt">type=</span><span class="st">&quot;b&quot;</span>,<span class="dt">xlab=</span><span class="st">&quot;year&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;allele frequency&quot;</span>, <span class="dt">xaxt=</span><span class="st">&quot;n&quot;</span>, <span class="dt">cex=</span><span class="fl">1.5</span>, <span class="dt">main=</span><span class="st">&quot;SNP # 906&quot;</span>)
<span class="kw">axis</span>(<span class="dt">side=</span><span class="dv">1</span>, <span class="dt">at=</span><span class="dv">1</span>:<span class="dv">6</span>, <span class="dt">lab=</span><span class="dv">2001</span>:<span class="dv">2006</span>)
<span class="kw">matplot</span>(freq399, <span class="dt">pch=</span><span class="kw">c</span>(<span class="st">&quot;c&quot;</span>,<span class="st">&quot;t&quot;</span>), <span class="dt">type=</span><span class="st">&quot;b&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;year&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;allele frequency&quot;</span>, <span class="dt">xaxt=</span><span class="st">&quot;n&quot;</span>, <span class="dt">cex=</span><span class="fl">1.5</span>, <span class="dt">main=</span><span class="st">&quot;SNP #399&quot;</span>)
<span class="kw">axis</span>(<span class="dt">side=</span><span class="dv">1</span>, <span class="dt">at=</span><span class="dv">1</span>:<span class="dv">6</span>, <span class="dt">lab=</span><span class="dv">2001</span>:<span class="dv">2006</span>)</code></pre>
<p><img src="DAPC_files/figure-html/unnamed-chunk-5-1.png" title="" alt="" width="960" /></p>
<p>This plot nicely illustrates the effect of mutation, followed by selection or drift in the seasonal H3N2 influenza virus.</p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>DAPC is a wonderful tool for exploring structure of populations based on PCA and DA without making assumptions of panmixia. Thus, this technique provides a robust alternative to Bayesian clustering methods like STRUCTURE <span class="citation">(Pritchard et al., 2000)</span> that should not be used for clonal or partially clonal populations.</p>
<p>DAPC analysis is inherently interactive and cannot be scripted <em>a priori</em>. Please refer to the <a href="http://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf">vignette</a> written by Thibaut Jombart for a more interactive analysis.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!--------------->
<div class="references">
<p>Grünwald NJ., Goss EM. 2011. Evolution and population genetics of exotic and re-emerging pathogens: Novel tools and approaches. <em>Annual Review of Phytopathology</em> 49:249–267. Available at: <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto" class="uri">http://www.annualreviews.org/doi/abs/10.1146/annurev-phyto-072910-095246?journalCode=phyto</a></p>
<p>Jombart T., Devillard S., Balloux F. 2010. Discriminant analysis of principal components: A new method for the analysis of genetically structured populations. <em>BMC genetics</em> 11:94. Available at: <a href="http://www.biomedcentral.com/1471-2156/11/94" class="uri">http://www.biomedcentral.com/1471-2156/11/94</a></p>
<p>Pritchard JK., Stephens M., Donnelly P. 2000. Inference of population structure using multilocus genotype data. <em>Genetics</em> 155:945–959. Available at: <a href="http://www.genetics.org/content/155/2/945.abstract" class="uri">http://www.genetics.org/content/155/2/945.abstract</a></p>
</div>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
