```{r,message=FALSE,echo=FALSE}
  library("knitcitations")
  library("knitr")
  cite_options(citation_format = "pandoc", max.names = 3, style = "html", hyperlink = "to.doc")
  bib <- read.bibtex("bibtexlib.bib")
  opts_chunk$set(tidy = FALSE)
```
---
title: "Data sets used in this primer"
---

AFLP (diploid, clonal): `Aeut`
----

This data is for a population of the diploid plant pathogen *Aphanomyces
euteiches* using AFLP as a marker system 
`r citep(bib[c("grunwald2006hierarchical")])`. A total of 187 pathogen isolates
were sampled hierarchically in two regions in Oregon (N = 97) and Washington (N
= 90). The alleles observed were treated as dominant markers with presence or
absence. Analysis included clone-correction, genotypic diversity, linkage
disequilibrium using the index of association, dendrograms based on Nei's
genetic distance, and AMOVA.


Microsatellite (haploid, clonal): `monpop`
----

This is microsatellite data for a population of the haploid plant pathogen
*Monilinia fructicola* that causes disease within peach tree canopies 
`r citep(bib[c("everhart2014finescale")])`.  Entire populations within trees
were sampled across 3 years (2009, 2010, and 2011) in a total of four trees,
where one tree was sampled in all three years, for a total of 6 within-tree
populations.  Within each year, samples in the spring were taken from affected
blossoms (termed "BB" for blossom blight) and in late summer from affected
fruits (termed "FR" for fruit rot).  There are a total of 694 isolates with 65
to 173 isolates within each canopy population that were characterized using a
set of 13 microsatellite markers.

Microsatellite (diploid, sexual): `microbov`
----

This data set contains 704 bovine samples over 30 microsatellite loci for 15
breeds, 2 species and 2 countries. From `r citep(bib["laloe2007consensus"])`.

Microsatellite (diploid, sexual): `nancycats`
----

This is a data set of 217 cats *Felis catus* L. from Nancy, France genotyped over 
9 microsatellite loci. They have been divided into 17 populations with spatial
coordinates in the `@other` slot. It comes from the *adegenet* package without
a published reference. Likely, the source paper is [here](http://dx.doi.org/10.1016/j.mambio.2008.01.001).

Microsatellite (triploid, sexual/clonal): `Pinf`
----

This microsatellite data comes from a larger data set of populations of
*Phytophthora infestans* `r citep(bib["goss2014irish"])`. It contains 86
individuals representing 72 multilocus genotypes that have been genotyped over
11 loci. They are grouped by continent and country. This data set is used to
demonstrate clone correction and linkage disequilibrium.

Microsatellite (diploid, clonal): `Pram`
----

This data set contains populations of *Phytophthora ramorum* from Nurseries in
California and Oregon `r citep(bib["goss2009population"])` and Forests in Curry
County, Oregon from 2001 to 2014 `r citep(bib['kamvar2015spatial'])`. There are
729 samples representing 98 multilocus genotypes genotyped over 5 microsatellite
loci.


SNP (haploid, clonal): `H3N2`
----

This is a SNP data set from the hemaglutinin segment of the H3N2 strain of
seasonal influenza contained within the *adegenet* package. It contains 1902
samples genotyped over 125 SNP markers. It contains a wealth of information in
the `@other` slot including year isolated, month isolated, country of origin,
and more. This data set is utilized to demonstrate the DAPC function in
*adegenet*.


GBS (diploid): `Prubi_gbs`
----


```{r child = 'data_gbs.Rmd'}
```


Genomic (diploid): `pinfsc50`
----


```{r child = 'data_genome.Rmd'}
```


References
----------
