<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>3: Data Preparation</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/readable.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Hierarchies.html">5. Population hierarchies and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="AMOVA.html">10. AMOVA</a></li>
               <li><a href="DAPC.html">11. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="Function_Glossary.html">A2. Function glossary</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">3: Data Preparation</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#importing-data">Importing data</a></li>
<li><a href="#genalex-data-format">GenAlEx data format</a></li>
<li><a href="#population-hierarchies">Population hierarchies</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>This section will briefly go over the basics of data import into <em>poppr</em>. For this section, we will focus on the GenAlEx format. Other formats are supported and details are given in the R help page for the <em>adegenet</em> function <code>import2genind</code>. We will show examples of haploid, diploid, and polyploid data sets and show you how you can format your data if it’s grouped into multiple hierarchical levels.</p>
<div id="importing-data" class="section level2">
<h2>Importing data</h2>
<div id="genalex-formatted-data" class="section level3">
<h3>GenAlEx formatted data</h3>
<p>To start simple, let’s examine the input file for the <code>monpop</code> data set containing 694 isolates of the plant pathogen <em>Monilinia fructicola</em> genotyped over 13 haploid loci <span class="citation">(Everhart &amp; Scherm, 2015)</span>. The data set is called <strong>monpop.csv</strong>.</p>
<blockquote>
<p><a class="btn btn-info btn-sm" target = "blank" href="https://github.com/grunwaldlab/Population_Genetics_in_R/blob/master/monpop.csv" role="button">view</a> or <a class="btn btn-primary btn-sm" target = "blank" href="http://grunwaldlab.github.io/Population_Genetics_in_R/monpop.csv" role="button"> download</a> monpop.csv</p>
</blockquote>
<p>To import GenAlEx formatted data into <em>poppr</em>, you should use the function <code>read.genalex</code>. Below is an example using <strong>monpop.csv</strong>.</p>
<pre class="r"><code>library(&quot;poppr&quot;)
monpop &lt;- read.genalex(&quot;~/Downloads/monpop.csv&quot;)
monpop</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      1 hierarchical level - Pop
##     12 populations defined - 7_09_BB 26_09_BB 26_09_FR ... 45_10_FR 
## 26_11_BB 26_11_FR</code></pre>
<blockquote>
<p>There are other options for genalex formatted data, such as xy coordinates. We won’t cover the other options, but type <code>help(&quot;read.genalex&quot;)</code> in your R console for details.</p>
</blockquote>
</div>
<div id="other-data-formats" class="section level3">
<h3>Other data formats</h3>
<p>Being based off of <em>adegenet</em>, users can import data from the following formats</p>
<ul>
<li><strong>FSTAT</strong> (file.dat)</li>
<li><strong>GENETIX</strong> (file.gtx)</li>
<li><strong>GENEPOP</strong> (file.gen)</li>
<li><strong>STRUCTURE</strong> (file.str)</li>
</ul>
<p>The <em>adegenet</em> function <code>import2genind</code> will import all of these formats. If you have sequence data, you can use the <code>read.FASTA</code> function from the <em>ape</em> package. If your data is in any other format, type <code>help(&quot;df2genind&quot;)</code> for guidance.</p>
<p>We will continue discussing the GenAlEx format below.</p>
</div>
</div>
<div id="genalex-data-format" class="section level2">
<h2>GenAlEx data format</h2>
<p>GenAlEx is a very popular add-on for Microsoft Excel. It is relatively easy to use because of its familiar, menu-driven interface. It also gives the user the option to include information on population groupings, regional groupings, and xy coordinates. The flexibility of this format made it a clear choice for import into <em>poppr</em>.</p>
<p>The data format is standard in that individuals are defined in the rows and loci are defined in the columns. The first two rows are reserved for metadata and the first two columns are reserved for the individual names and population names, respectively. The examples we will be using are codominant data, but AFLP data is also accepted.</p>
<div id="haploids" class="section level3">
<h3>Haploids</h3>
<p>Below is what the monpop.csv data look like. Highlighted in red is how missing data should be coded for codominant markers. Highlighted in blue are the parts of the metadata rows used by <em>poppr</em>. These three numbers represent:</p>
<table align="left" cellpadding=20 border=2>
  <thead>
    <tr>
      <th class="col-xs-3">
A
</th>
      <th class="col-xs-3">
B
</th>
      <th class="col-xs-3">
C
</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="text-center">
loci
</td>
      <td class="text-center">
samples
</td>
      <td class="text-center">
populations
</td>
    </tr>
  </tbody>
</table>

<p><img src="images/monpop.png" alt="monpop" /></p>
<p>The columns of the metadata beyond those three rows define the number of individuals contained within each population. Since this data is redundant with the second column, it is not necessary. Notice, also, that the second column, reserved for the population assignments, has a pattern of underscores in the populations. This will be important at the end of this section. Below is a modified version of the input format that should make it easier to format.</p>
<p><img src="images/monpop_ideal.png" alt="monpop_ideal" /></p>
<p>Highlighted in blue is the cell that defines the number of columns highlighted in red. If we set this number to <code>1</code>, then we do not have to enter in any information in those columns. Try it for yourself.</p>
</div>
<div id="diploids" class="section level3">
<h3>Diploids</h3>
<p>Diploid data is only different in the fact that you will have two alleles represented at each locus. This is coded such that each allele is in a separate column so that your genetic data will take up 2 <span class="math">\(\times\)</span> (number of loci) columns. Below is an example of the <code>nancycats</code> data set (from the <em>adegenet</em> package), exported like above. Highlighted in blue and red are the first and second loci, respectively.</p>
<p><img src="images/diploid.png" alt="diploid" /></p>
</div>
<div id="polyploids" class="section level3">
<h3>Polyploids</h3>
<p>GenAlEx does not handle polyploids, but since poppr can do it, we have set up a scheme to allow import of polyploids via this format. The limitation is that all of your loci have to have the same observed ploidy. Below is the example of <em>Phytophthora infestans</em> in the data set <code>Pinf</code> where some genotypes had observed tetraploid loci <span class="citation">(Goss et al., 2014)</span>.</p>
<p><img src="images/tetraploid.png" alt="tetraploid" /></p>
<p>Highlighted in blue is the first locus and highlighted in red are two samples at that locus, an observed diploid and observed triploid. Note the extra zeroes needed to make the genotype tetraploid.</p>
</div>
</div>
<div id="population-hierarchies" class="section level2">
<h2>Population hierarchies</h2>
<p>A hierarchical sampling approach is necessary to infer structure of populations in space or time. <em>Poppr</em> facilitates definition of hierarchical data by concatenating the different hierarchical levels into a single column by a common separator (“_” by default). Here’s an example of the three hierarchical levels of the <code>monpop</code> data set introduced above:</p>
<table border=2 cellpadding=20 align=center>
<tr> <th class="col-xs3"> 
Tree
</th> <th class="col-xs3"> 
Year
</th> <th class="col-xs3"> 
Symptom
</th>  </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
  <tr> <td align="center"> 
7
</td> <td align="center"> 
9
</td> <td align="center"> 
BB
</td> </tr>
</table>

<p>resulting in definition for use in <em>poppr</em> as:</p>
<!-- html table generated in R 3.1.2 by xtable 1.7-4 package -->
<!-- Fri Jan  9 20:12:57 2015 -->
<table border=2 cellpadding=20 align=center>
<tr> <th class="col-xs3"> 
Tree_Year_Symptom
</th>  </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
  <tr> <td align="center"> 
7_9_BB
</td> </tr>
</table>



<blockquote>
<p>If you do not want to concatenate your population factors into one row, save them as a separate csv file and use the functions <code>read.table</code> and <code>sethierarchy</code> after you’ve imported your genetic data to acheive the same result.</p>
</blockquote>
<p>The data imported this way can be accessed with the function <code>splithierarchy</code>:</p>
<pre class="r"><code># using monpop from above
monpop # Before (One level, Pop)</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      1 hierarchical level - Pop
##     12 populations defined - 7_09_BB 26_09_BB 26_09_FR ... 45_10_FR 
## 26_11_BB 26_11_FR</code></pre>
<pre class="r"><code>splithierarchy(monpop) &lt;- ~Tree/Year/Symptom
monpop # After (Three distinct levels)</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 hierarchical levels - Tree Year Symptom
##     12 populations defined - 7_09_BB 26_09_BB 26_09_FR ... 45_10_FR 
## 26_11_BB 26_11_FR</code></pre>
<p>This is detailed in <a href="Population_Hierarchies.html">our chapter on population hierarchies</a>.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!------>


<div class="references">
<p>Everhart S., Scherm H. 2015. Fine-scale genetic structure of <em>Monilinia fructicola</em> during brown rot epidemics within individual peach tree canopies. <em>Phytopathology</em> ##:in press.</p>
<p>Goss E., Tabima J., Cooke D., Restrepo S., Fry W., Forbes G., Fieland V., Cardenas M., Grünwald N. 2014. The irish potato famine pathogen <em>Phytophthora infestans</em> originated in central mexico rather than the andes. <em>Proceedings of the National Academy of Sciences</em> 111:8791–8796. Available at: <a href="http://www.pnas.org/content/early/2014/05/29/1401884111.abstract">http://www.pnas.org/content/early/2014/05/29/1401884111.abstract</a></p>
</div>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
