<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Function_Glossary.utf8.md</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="quality_control.html">Quality control</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">A2: Function glossary</h1>

</div>


<!---To quit R you can either use the Rstudio>Quit Rstudio pulldown menu command or execute ⌘+Q (OS X) or CTRL+Q (PC). 

Navigation in R
----

---->
<!--```{r, eval=FALSE}
getwd()        # this command will print the current working directory
ls()           # list objects in the current workspace
help(options)  # learn about available options
history()      # viewing your command history
data(*dsname*) # access builtin database *dsname*; must have pacakge loaded containing it
q()            # quit your R session

# default for saving history is ".Rhistory"
savehistory(file="myfile")  # save your command history for whole session
loadhistory(file= "myfile") # recall command history save

# Package specific information
installed.packages()[,c("Package","Version")] #list packages & versions installed
browseVignettes()                     # see vignettes from all packages

#Help on functions
help(functionname)     # Display documentation for a function
args(functionname)     # What are the arguments for a given function
example(functionname)  # see examples of function calls
```
-->
<div id="navigation-in-r" class="section level2">
<h2>Navigation in R</h2>
<p>Our primer is heavily based on the <em>poppr</em> and <em>adegenet</em> packages. To get help on any of their functions type a question mark before the empty function call as in:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">?mlg <span class="co"># open the R documentiaon of the function mlg</span></a></code></pre></div>
<p>Other information can be obtained with the following commands.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">browseVignettes</span>(<span class="dt">package =</span> <span class="st">&#39;poppr&#39;</span>) <span class="co"># see vignettes for poppr</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">vignette</span>(<span class="st">&#39;poppr_manual&#39;</span>)           <span class="co"># see manual for poppr</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">help</span>(<span class="dt">package=</span><span class="st">&quot;poppr&quot;</span>)              <span class="co"># shows all functions contained in poppr package</span></a></code></pre></div>
</div>
<div id="poppr-functions" class="section level2">
<h2><em>Poppr</em> functions</h2>
<div id="importexport" class="section level3">
<h3>Import/Export</h3>
<table>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>getfile</code></td>
<td>Provides a quick GUI to grab files for import</td>
</tr>
<tr class="even">
<td><code>read.genalex</code></td>
<td>Read <em>GenAlEx</em> formatted csv files to a genind object</td>
</tr>
<tr class="odd">
<td><code>genind2genalex</code></td>
<td>Converts genind objects to <em>GenAlEx</em> formatted csv files</td>
</tr>
</tbody>
</table>
</div>
<div id="manipulation" class="section level3">
<h3>Manipulation</h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>as.genclone</code></td>
<td>Converts genind objects to genclone objects</td>
</tr>
<tr class="even">
<td><code>setPop</code></td>
<td>Set the population using defined strata</td>
</tr>
<tr class="odd">
<td><code>splitStrata</code></td>
<td>Split concatenated strata imported as a population</td>
</tr>
<tr class="even">
<td><code>strata</code></td>
<td>Define or extract population strata of a genind object</td>
</tr>
<tr class="odd">
<td><code>addStrata</code></td>
<td>Add a vector or data frame to an existing strata</td>
</tr>
<tr class="even">
<td><code>nameStrata</code></td>
<td>Rename a population strata</td>
</tr>
<tr class="odd">
<td><code>missingno</code></td>
<td>Handles missing data</td>
</tr>
<tr class="even">
<td><code>clonecorrect</code></td>
<td>Clone censor at a specified population stratum</td>
</tr>
<tr class="odd">
<td><code>informloci</code></td>
<td>Detects and removes phylogenetically uninformative loci</td>
</tr>
<tr class="even">
<td><code>popsub</code></td>
<td>Subsets genind objects by population</td>
</tr>
<tr class="odd">
<td><code>shufflepop</code></td>
<td>Shuffles genotypes at each locus using four different shuffling algorithms</td>
</tr>
<tr class="even">
<td><code>recode_polyploids</code></td>
<td>Recodes an imported polyploid data set to observed allelic frequencies</td>
</tr>
</tbody>
</table>
</div>
<div id="distances" class="section level3">
<h3>Distances</h3>
<table>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>bruvo.dist</code></td>
<td>Bruvo’s distance</td>
</tr>
<tr class="even">
<td><code>diss.dist</code></td>
<td>Absolute genetic distance (see <code>provesti.dist</code>)</td>
</tr>
<tr class="odd">
<td><code>nei.dist</code></td>
<td>Nei’s 1978 genetic distance</td>
</tr>
<tr class="even">
<td><code>rogers.dist</code></td>
<td>Rogers’ euclidean distance</td>
</tr>
<tr class="odd">
<td><code>reynolds.dist</code></td>
<td>Reynolds’ coancestry distance</td>
</tr>
<tr class="even">
<td><code>edwards.dist</code></td>
<td>Edwards’ angular distance</td>
</tr>
<tr class="odd">
<td><code>provesti.dist</code></td>
<td>Provesti’s absolute genetic distance</td>
</tr>
</tbody>
</table>
</div>
<div id="bootstrapping" class="section level3">
<h3>Bootstrapping</h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>aboot</code></td>
<td>Creates a bootstrapped dendrogram for any distance measure</td>
</tr>
<tr class="even">
<td><code>bruvo.boot</code></td>
<td>Produces dendrograms with bootstrap support based on Bruvo’s distance</td>
</tr>
</tbody>
</table>
</div>
<div id="analysis" class="section level3">
<h3>Analysis</h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>poppr.amova</code></td>
<td>Analysis of Molecular Variance (as implemented in ade4)</td>
</tr>
<tr class="even">
<td><code>ia</code></td>
<td>Calculates the index of association</td>
</tr>
<tr class="odd">
<td><code>mlg</code></td>
<td>Calculates the number of multilocus genotypes</td>
</tr>
<tr class="even">
<td><code>mlg.crosspop</code></td>
<td>Finds all multilocus genotypes that cross populations</td>
</tr>
<tr class="odd">
<td><code>mlg.table</code></td>
<td>Returns a table of populations by multilocus genotypes</td>
</tr>
<tr class="even">
<td><code>mlg.vector</code></td>
<td>Returns a vector of a numeric multilocus genotype assignment for each individual</td>
</tr>
<tr class="odd">
<td><code>mlg.id</code></td>
<td>Finds all individuals associated with each multilocus genotype</td>
</tr>
<tr class="even">
<td><code>poppr</code></td>
<td>Returns a diversity table by population</td>
</tr>
<tr class="odd">
<td><code>poppr.all</code></td>
<td>Returns a diversity table by population for all compatible files specified</td>
</tr>
<tr class="even">
<td><code>private_alleles</code></td>
<td>Tabulates the occurences of alleles that only occur in one population</td>
</tr>
<tr class="odd">
<td><code>locus_table</code></td>
<td>Create a table of summary statistics per locus</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization" class="section level3">
<h3>Visualization</h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>plot_poppr_msn</code></td>
<td>Plots minimum spanning networks produced in poppr with scale bar and legend</td>
</tr>
<tr class="even">
<td><code>greycurve</code></td>
<td>Helper to determine the appropriate parameters for adjusting the grey level for msn functions</td>
</tr>
<tr class="odd">
<td><code>bruvo.msn</code></td>
<td>Produces minimum spanning networks based off Bruvo’s distance colored by population</td>
</tr>
<tr class="even">
<td><code>poppr.msn</code></td>
<td>Produces a minimum spanning network for any pairwise distance matrix related to the data</td>
</tr>
<tr class="odd">
<td><code>info_table</code></td>
<td>Creates a heatmap representing missing data or observed ploidy</td>
</tr>
<tr class="even">
<td><code>genotype_curve</code></td>
<td>Creates a series of boxplots to demonstrate how many markers are needed to represent the diversity of your data</td>
</tr>
</tbody>
</table>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
