<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Genetic differentiation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="qc.html">Quality control</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genetic differentiation</h1>

</div>


<p>A fundamental question to most population studies is whether populations are diverse and whether this diversity is shared among the populations? To address the question of within population diversity geneticists typically report heterozygosity. This is the probability that two alleles randomly chosen from a population will be different <span class="citation">(Nei, 1973)</span>. Ecologists may know this as Simpson’s Index <span class="citation">(Simpson, 1949)</span>. To address differentiation population geneticists typically utilize <span class="math inline">\(F_{ST}\)</span> or one of its analogues. Population differentiation measured by <span class="math inline">\(F_{ST}\)</span> was originally proposed by Sewall Wright <span class="citation">(Wright, 1949)</span>. This was later extended to a method based on diversity by Masatoshi Nei <span class="citation">(Nei, 1973)</span>. As researchers applied these metrics to microsatellites, genetic markers with a large number of alleles, it became clear that Nei’s measure would not correctly range from zero to one, so Philip Hedrick proposed a correction <span class="citation">(Hedrick, 2005)</span>. More recently, Lou Jost proposed another alternative <span class="citation">(Jost, 2008)</span>. You can tell a topic is popular when so many variants of it are generated. And there are more variants than mentioned here. A nice discussion as to which measure may be appropriate for your data was posteed to teh Molecular Ecologist blog titled <a href="http://www.molecularecologist.com/2011/03/should-i-use-fst-gst-or-d-2/">should I use <span class="math inline">\(F_{ST}\)</span>, <span class="math inline">\(G_{ST}\)</span> or <span class="math inline">\(D\)</span>?</a>.</p>
<p>In <code>vcfR</code>, the function <code>genetic_diff()</code> was implemented to measure population diversity and differentiation. Because VCF data typically do not include population information we’ll have to supply it as a factor. The method ‘nei’ employed here is based on the methods reported by Hedrick <span class="citation">(Hedrick, 2005)</span>. The exception is that the heterozygosities are weighted by the number of alleles observed in each population. This was inspired by <code>hierfstat::pairwise.fst()</code> which uses the number of individuals observed in each population to weight the heterozygosities. By using the number of alleles observed instead of the number of individuals we remove an assumption about how many alleles each individual may contribute. That is, we should be able to accomodate samples of mixed ploidy.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vcfR)
<span class="kw">data</span>(vcfR_example)
pop &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">c</span>(<span class="st">&quot;us&quot;</span>, <span class="st">&quot;eu&quot;</span>, <span class="st">&quot;us&quot;</span>, <span class="st">&quot;af&quot;</span>, <span class="st">&quot;eu&quot;</span>, <span class="st">&quot;us&quot;</span>, <span class="st">&quot;mx&quot;</span>, <span class="st">&quot;eu&quot;</span>, <span class="st">&quot;eu&quot;</span>, <span class="st">&quot;sa&quot;</span>, <span class="st">&quot;mx&quot;</span>, <span class="st">&quot;sa&quot;</span>, <span class="st">&quot;us&quot;</span>, <span class="st">&quot;sa&quot;</span>, <span class="st">&quot;Pmir&quot;</span>, <span class="st">&quot;us&quot;</span>, <span class="st">&quot;eu&quot;</span>, <span class="st">&quot;eu&quot;</span>))
myDiff &lt;-<span class="st"> </span><span class="kw">genetic_diff</span>(vcf, <span class="dt">pops =</span> pop, <span class="dt">method =</span> <span class="st">&#39;nei&#39;</span>)
knitr::<span class="kw">kable</span>(<span class="kw">head</span>(myDiff[,<span class="dv">1</span>:<span class="dv">15</span>]))</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">CHROM</th>
<th align="left">POS</th>
<th align="right">Hs_af</th>
<th align="right">Hs_eu</th>
<th align="right">Hs_mx</th>
<th align="right">Hs_Pmir</th>
<th align="right">Hs_sa</th>
<th align="right">Hs_us</th>
<th align="right">Ht</th>
<th align="right">n_af</th>
<th align="right">n_eu</th>
<th align="right">n_mx</th>
<th align="right">n_Pmir</th>
<th align="right">n_sa</th>
<th align="right">n_us</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">2</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.5</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.0798611</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">246</td>
<td align="right">NaN</td>
<td align="right">0.0</td>
<td align="right">0.375</td>
<td align="right">NaN</td>
<td align="right">0.000</td>
<td align="right">0.50</td>
<td align="right">0.3512397</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">549</td>
<td align="right">NaN</td>
<td align="right">0.0</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
<td align="right">0.50</td>
<td align="right">0.4444444</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">668</td>
<td align="right">NaN</td>
<td align="right">0.5</td>
<td align="right">0.000</td>
<td align="right">NaN</td>
<td align="right">0.000</td>
<td align="right">0.50</td>
<td align="right">0.5000000</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">Supercontig_1.50</td>
<td align="left">765</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.1107266</td>
<td align="right">2</td>
<td align="right">12</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">Supercontig_1.50</td>
<td align="left">780</td>
<td align="right">0</td>
<td align="right">0.0</td>
<td align="right">0.000</td>
<td align="right">0.0</td>
<td align="right">0.375</td>
<td align="right">0.18</td>
<td align="right">0.1244444</td>
<td align="right">2</td>
<td align="right">8</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">10</td>
</tr>
</tbody>
</table>
<p>The function returns the chromosome and position of each variant as provided in the VCF data. This should allow you to align its output with the VCF data. The heterozygosities for each population are reported as well as the total heterozygosity, followed by the number of alleles observed in each population. Note that in some populations zero alleles were observed. Populations with zero alleles reported heterozygosities of ‘NaN’ because of this absence of data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(<span class="kw">head</span>(myDiff[,<span class="dv">16</span>:<span class="dv">19</span>]))</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">Gst</th>
<th align="right">Htmax</th>
<th align="right">Gstmax</th>
<th align="right">Gprimest</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.4782609</td>
<td align="right">0.7951389</td>
<td align="right">0.9475983</td>
<td align="right">0.5047085</td>
</tr>
<tr class="even">
<td align="right">NaN</td>
<td align="right">0.8057851</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="right">NaN</td>
<td align="right">0.6666667</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="right">NaN</td>
<td align="right">0.8125000</td>
<td align="right">NaN</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="right">1.0000000</td>
<td align="right">0.7543253</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="right">0.1160714</td>
<td align="right">0.8000000</td>
<td align="right">0.8625000</td>
<td align="right">0.1345756</td>
</tr>
</tbody>
</table>
<p>The remaining columns contain <span class="math inline">\(G_{ST}\)</span>, the maximum heterozygosity, the maximum <span class="math inline">\(G_{ST}\)</span> and finally <span class="math inline">\(G&#39;_{ST}\)</span>. The maximum heterozygosity and the maximum <span class="math inline">\(G_{ST}\)</span> are intermediary values used to calculate <span class="math inline">\(G&#39;_{ST}\)</span>. They are typically not reported but provide values to help validate that <span class="math inline">\(G&#39;_{ST}\)</span> was calculated correctly. Note that the populations that had zero alleles, and therefore a heterozygosity of ‘NaN’, contributed to <span class="math inline">\(G_{ST}\)</span>s that were also ‘NaN’. To avoid this you may want to consider omitting populations with a small sample size or that contain a large amount of missing data.</p>
<p>We now have information for each variant in the VCF data. Because this is typically a large quantity of information, we’ll want to summarize it. One way is to take averages of the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(<span class="kw">round</span>(<span class="kw">colMeans</span>(myDiff[,<span class="kw">c</span>(<span class="dv">3</span>:<span class="dv">9</span>,<span class="dv">16</span>,<span class="dv">19</span>)], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>), <span class="dt">digits =</span> <span class="dv">3</span>))</code></pre></div>
<table>
<tbody>
<tr class="odd">
<td align="left">Hs_af</td>
<td align="right">0.176</td>
</tr>
<tr class="even">
<td align="left">Hs_eu</td>
<td align="right">0.188</td>
</tr>
<tr class="odd">
<td align="left">Hs_mx</td>
<td align="right">0.168</td>
</tr>
<tr class="even">
<td align="left">Hs_Pmir</td>
<td align="right">0.052</td>
</tr>
<tr class="odd">
<td align="left">Hs_sa</td>
<td align="right">0.198</td>
</tr>
<tr class="even">
<td align="left">Hs_us</td>
<td align="right">0.155</td>
</tr>
<tr class="odd">
<td align="left">Ht</td>
<td align="right">0.247</td>
</tr>
<tr class="even">
<td align="left">Gst</td>
<td align="right">0.595</td>
</tr>
<tr class="odd">
<td align="left">Gprimest</td>
<td align="right">0.632</td>
</tr>
</tbody>
</table>
<p>Another way to summarize data is to use violin plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(reshape2)
<span class="kw">library</span>(ggplot2)

dpf &lt;-<span class="st"> </span><span class="kw">melt</span>(myDiff[,<span class="kw">c</span>(<span class="dv">3</span>:<span class="dv">8</span>,<span class="dv">19</span>)], <span class="dt">varnames=</span><span class="kw">c</span>(<span class="st">&#39;Index&#39;</span>, <span class="st">&#39;Sample&#39;</span>), <span class="dt">value.name =</span> <span class="st">&#39;Depth&#39;</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## No id variables; using all as measure variables</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dpf, <span class="kw">aes</span>(<span class="dt">x=</span>variable, <span class="dt">y=</span>Depth)) +<span class="st"> </span><span class="kw">geom_violin</span>(<span class="dt">fill=</span><span class="st">&quot;#2ca25f&quot;</span>, <span class="dt">adjust =</span> <span class="fl">1.2</span>)
p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)
p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)
p &lt;-<span class="st"> </span>p +<span class="st"> </span><span class="kw">theme_bw</span>()
p</code></pre></div>
<p><img src="genetic_differentiation_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<div id="refs" class="references">
<div id="ref-hedrick2005standardized">
<p>Hedrick PW. 2005. A standardized genetic differentiation measure. <em>Evolution</em> 59:1633–1638. Available at: <a href="http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x" class="uri">http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x</a></p>
</div>
<div id="ref-jost2008gst">
<p>Jost L. 2008. <span class="math inline">\(G_{ST}\)</span> And its relatives do not measure differentiation. <em>Molecular Ecology</em> 17:4015–4026. Available at: <a href="http://dx.doi.org/10.1111/j.1365-294X.2008.03887.x" class="uri">http://dx.doi.org/10.1111/j.1365-294X.2008.03887.x</a></p>
</div>
<div id="ref-nei1973analysis">
<p>Nei M. 1973. Analysis of gene diversity in subdivided populations. <em>Proceedings of the National Academy of Sciences</em> 70:3321–3323. Available at: <a href="http://www.pnas.org/content/70/12/3321.abstract" class="uri">http://www.pnas.org/content/70/12/3321.abstract</a></p>
</div>
<div id="ref-simpson1949measurement">
<p>Simpson EH. 1949. Measurement of diversity. <em>Nature</em> 163:688. Available at: <a href="http://dx.doi.org/10.1038/163688a0" class="uri">http://dx.doi.org/10.1038/163688a0</a></p>
</div>
<div id="ref-wright1949genetical">
<p>Wright S. 1949. The genetical structure of populations. <em>Annals of Eugenics</em> 15:323–354. Available at: <a href="http://dx.doi.org/10.1111/j.1469-1809.1949.tb02451.x" class="uri">http://dx.doi.org/10.1111/j.1469-1809.1949.tb02451.x</a></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
