<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>7: Genotypic richness, diversity, and evenness</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/readable.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Strata.html">5. Population strata and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="Minimum_Spanning_Networks.html">10. Minimum Spanning Networks</a></li>
               <li><a href="AMOVA.html">11. AMOVA</a></li>
               <li><a href="DAPC.html">12. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="funpendix.html">A2. Function glossary</a></li>
            <li><a href="Intro_to_R.html">A3. Introduction to R</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">7: Genotypic richness, diversity, and evenness</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#genotypic-richness">Genotypic richness</a></li>
<li><a href="#genotypic-diversity">Genotypic diversity</a></li>
<li><a href="#genotypic-evenness">Genotypic evenness</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>In Chapter 4, we introduced basic summary statistics that can be calculated using <em>poppr</em>. For this chapter, we want to specifically focus on how to evaluate genotypic richness, diversity, and evenness in your data. In this example, we’ll examine the monpop microsatellite data for 13 loci of 694 individuals of the haploid fungal pathogen <em>Monilinia fructicola</em> that infects peach flowers and fruits in commercial orchards.</p>
<p>For this example, we can explore the hypothesis that the population that infects flowers and yields blighted blossoms (BB) is from a more genetically diverse pool (being a product of overwintering, sexual recombinants), than the population that infects and causes fruit rots (FR), which is likely a product of asexual production from existing infections in the orchard. As such, we can ask the following questions:</p>
<ul>
<li>Is genotypic richness of BB populations higher than for FR populations?</li>
<li>Is genotypic diversity of BB populations higher for FR populations?</li>
<li>Is genotypic evenness higher for BB populations than for FR populations?</li>
</ul>
<p>For the analysis, we need to read in the data, specify the stratifications in the data, and then set the stratification to symptom so that we can calculate genotypic richness, diversity, and evenness for BB as compared to FR for the entire data set:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">data</span>(monpop)
<span class="kw">splitStrata</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree/Year/Symptom
<span class="kw">setPop</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Symptom
monpop</code></pre>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree Year Symptom
##      2 populations defined - BB FR</code></pre>
<p>To calculate genotypic richness, diversity, and evenness, we can use the <code>poppr</code> function:</p>
<pre class="sourceCode r"><code class="sourceCode r">(monpop_diversity &lt;-<span class="st"> </span><span class="kw">poppr</span>(monpop))</code></pre>
<pre><code>## | BB 
## | FR 
## | Total</code></pre>
<pre><code>##     Pop   N MLG eMLG   SE    H    G lambda   E.5  Hexp    Ia  rbarD   File
## 1    BB 113  94 94.0 0.00 4.40 61.7  0.984 0.755 0.584 0.591 0.0493 monpop
## 2    FR 581 191 66.6 4.17 4.58 53.4  0.981 0.543 0.588 0.809 0.0679 monpop
## 3 Total 694 264 73.6 4.33 4.89 65.0  0.985 0.486 0.589 0.729 0.0611 monpop</code></pre>
<p>This shows us several summary statistics:</p>
<table>
<thead>
<tr class="header">
<th align="left">Abbreviation</th>
<th align="left">Statistic</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>Pop</code></td>
<td align="left">Population name.</td>
</tr>
<tr class="even">
<td align="left"><code>N</code></td>
<td align="left">Number of individuals observed.</td>
</tr>
<tr class="odd">
<td align="left"><code>MLG</code></td>
<td align="left">Number of multilocus genotypes (MLG) observed.</td>
</tr>
<tr class="even">
<td align="left"><code>eMLG</code></td>
<td align="left">The number of expected MLG at the smallest sample size ≥ 10 based on rarefaction <span class="citation">(Hurlbert, 1971)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><code>SE</code></td>
<td align="left">Standard error based on eMLG.</td>
</tr>
<tr class="even">
<td align="left"><code>H</code></td>
<td align="left">Shannon-Wiener Index of MLG diversity <span class="citation">(Shannon, 2001)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><code>G</code></td>
<td align="left">Stoddart and Taylor’s Index of MLG diversity <span class="citation">(Stoddart &amp; Taylor, 1988)</span>.</td>
</tr>
<tr class="even">
<td align="left"><code>lambda</code></td>
<td align="left">Simpson’s Index <span class="citation">(Simpson, 1949)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><code>E.5</code></td>
<td align="left">Evenness, <span class="math">\(E_5\)</span> <span class="citation">(Pielou, 1975; Ludwig &amp; Reynolds, 1988; Grünwald et al., 2003)</span>.</td>
</tr>
<tr class="even">
<td align="left"><code>Hexp</code></td>
<td align="left">Nei’s unbiased gene diversity <span class="citation">(Nei, 1978)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><code>Ia</code></td>
<td align="left">The index of association, <span class="math">\(I_A\)</span> <span class="citation">(Brown, Feldman &amp; Nevo, 1980; Smith et al., 1993)</span>.</td>
</tr>
<tr class="even">
<td align="left"><code>rbarD</code></td>
<td align="left">The standardized index of association, <span class="math">\(\bar{r}_d\)</span> <span class="citation">(Agapow &amp; Burt, 2001)</span>.</td>
</tr>
</tbody>
</table>
<div id="genotypic-richness" class="section level2">
<h2>Genotypic richness</h2>
<p>The number of observed <span class="math">\(MLGs\)</span> is equivalent to genotypic richness. We expect that the BB population would have a higher genotypic richness than the FR population. However, looking at the raw number of MLGs for each symptom type, it shows us the opposite: there are 94 MLGs for BB and 191 MLGs for FR. This discrepancy has to do with the sample size differences, namely <span class="math">\(N = 113\)</span> for BB and <span class="math">\(N = 581\)</span> for FR. A more appropriate comparison is the <span class="math">\(eMLG\)</span> value, which is an approximation of the number of genotypes that would be expected at the largest, shared sample size (<span class="math">\(N = 113\)</span>) based on rarefaction. For BB (<span class="math">\(N = 113\)</span>) the <span class="math">\(eMLG = 94\)</span> and for FR (where <span class="math">\(N\)</span> is set to 113) the <span class="math">\(eMLG\)</span> = 66.6. Thus, genotypic richness is indeed higher in the BB populations than the FR population when considering equal sample sizes.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;vegan&quot;</span>)
mon.tab &lt;-<span class="st"> </span><span class="kw">mlg.table</span>(monpop, <span class="dt">plot =</span> <span class="ot">FALSE</span>)
min_sample &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">rowSums</span>(mon.tab))
<span class="kw">rarecurve</span>(mon.tab, <span class="dt">sample =</span> min_sample, <span class="dt">xlab =</span> <span class="st">&quot;Sample Size&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Expected MLGs&quot;</span>)
<span class="kw">title</span>(<span class="st">&quot;Rarefaction of Fruit Rot and Blossom Blight&quot;</span>)</code></pre>
<p><img src="Genotypic_EvenRichDiv_files/figure-html/rarecurve-1.png" title="" alt="" width="700px" /></p>
</div>
<div id="genotypic-diversity" class="section level2">
<h2>Genotypic diversity</h2>
<p>Diversity measures incorporate both genotypic richness and abundance. There are three measures of genotypic diversity employed by <em>poppr</em>, the Shannon-Wiener index (H), Stoddart and Taylor’s index (G), and Simpson’s index (lambda). In our example, comparing the diversity of BB to FR shows that H is greater for FR (4.58 vs. 4.4), but G is lower (53.4 vs. 61.7). Thus, our expectation that diversity is lower for FR than BB is rejected in the case of H, which is likely due to the sensitivity of the Shannon-Wiener index to genotypic richness in the uneven sample sizes, and accepted in the case of G. To be fair, the sample size used to calculate these diversity measures is different and is therefore not an appropriate comparison.</p>
<p>For an easier statistic to grasp, we have included the Simpson index, which is simply one minus the sum of squared genotype frequencies. This measure provides an estimation of the probability that two randomly selected genotypes are different and scales from 0 (no genotypes are different) to 1 (all genotypes are different). In the data above, we can see that lambda is just barely higher in BB than FR (0.984 vs. 0.981). Since this might be an artifact of sample size, we can explore a correction of Simpson’s index for sample size by multiplying lambda by <span class="math">\(N/(N - 1)\)</span>. Since R is vectorized, we can do this for all of our populations at once:</p>
<pre class="sourceCode r"><code class="sourceCode r">N      &lt;-<span class="st"> </span>monpop_diversity$N      <span class="co"># number of samples</span>
lambda &lt;-<span class="st"> </span>monpop_diversity$lambda <span class="co"># Simpson&#39;s index</span>
(N/(N -<span class="st"> </span><span class="dv">1</span>)) *<span class="st"> </span>lambda              <span class="co"># Corrected Simpson&#39;s index</span></code></pre>
<pre><code>## [1] 0.9925727 0.9829604 0.9860399</code></pre>
<p>Now we can see that, even after correction, Simpson’s index is still higher for BB.</p>
<blockquote>
<p>You try it! Can you calculate the clonal fraction for each population (1 - (MLG/N))?</p>
</blockquote>
</div>
<div id="genotypic-evenness" class="section level2">
<h2>Genotypic evenness</h2>
<p>Evenness is a measure of the distribution of genotype abundances, wherein a population with equally abundant genotypes yields a value equal to 1 and a population dominated by a single genotype is closer to zero. In the example above, the BB population has <span class="math">\(E.5 = 0.755\)</span> and the FR population has <span class="math">\(E.5 = 0.543\)</span> . This indicates that the MLGs observed in the BB population are closer to equal abundance than those in the FR population. Indeed, when we look at a distribution of the <span class="math">\(MLGs\)</span> for each symptom type it shows us there are many more unique BB symptoms as compared to the FR symptoms.</p>
<pre class="sourceCode r"><code class="sourceCode r">mon.tab &lt;-<span class="st"> </span><span class="kw">mlg.table</span>(monpop)</code></pre>
<p><img src="Genotypic_EvenRichDiv_files/figure-html/symptom_table-1.png" title="" alt="" width="1000px" /></p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>Calculating measures of genotypic richness, diversity, and evenness is straightforward to do in <em>poppr</em>. In our example, we were able to perform these calculations with one command. However, the ease of calculating these measures is not an indication of the ease of interpretation, particularly when it comes to measures of diversity. There are a large number of diversity measures available and the measures provided here are those we found most useful.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!--------->
<div class="references">
<p>Agapow P-M., Burt A. 2001. Indices of multilocus linkage disequilibrium. <em>Molecular Ecology Notes</em> 1:101–102. Available at: <a href="http://onlinelibrary.wiley.com/doi/10.1046/j.1471-8278.2000.00014.x/abstract" class="uri">http://onlinelibrary.wiley.com/doi/10.1046/j.1471-8278.2000.00014.x/abstract</a></p>
<p>Brown AHD., Feldman MW., Nevo E. 1980. Multilocus structure of natural populations of <em>Hordeum spontaneum</em>. <em>Genetics</em> 96:523–536. Available at: <a href="http://www.genetics.org/content/96/2/523" class="uri">http://www.genetics.org/content/96/2/523</a></p>
<p>Grünwald NJ., Goodwin SB., Milgroom MG., Fry WE. 2003. Analysis of genotypic diversity data for populations of microorganisms. <em>Phytopathology</em> 93:738–746. Available at: <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738" class="uri">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738</a></p>
<p>Hurlbert SH. 1971. The nonconcept of species diversity: A critique and alternative parameters. <em>Ecology</em> 52:577–586. Available at: <a href="http://www.jstor.org/discover/10.2307/1934145?uid=3739856&amp;uid=2&amp;uid=4&amp;uid=3739256&amp;sid=21103760010461" class="uri">http://www.jstor.org/discover/10.2307/1934145?uid=3739856&amp;uid=2&amp;uid=4&amp;uid=3739256&amp;sid=21103760010461</a></p>
<p>Ludwig JA., Reynolds JF. 1988. <em>Statistical ecology: A primer in methods and computing</em>. Wiley.com.</p>
<p>Nei M. 1978. Estimation of average heterozygosity and genetic distance from a small number of individuals. <em>Genetics</em> 89:583–590. Available at: <a href="http://www.genetics.org/content/89/3/583.abstract" class="uri">http://www.genetics.org/content/89/3/583.abstract</a></p>
<p>Pielou EC. 1975. <em>Ecological diversity</em>. Wiley New York.</p>
<p>Shannon CE. 2001. A mathematical theory of communication. <em>ACM SIGMOBILE Mobile Computing and Communications Review</em> 5:3–55. Available at: <a href="http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf" class="uri">http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf</a></p>
<p>Simpson E. 1949. Measurement of diversity. <em>Nature</em> 163:688.</p>
<p>Smith JM., Smith NH., O’Rourke M., Spratt BG. 1993. How clonal are bacteria. <em>Proceedings of the National Academy of Sciences</em> 90:4384–4388. Available at: <a href="http://www.pnas.org/content/90/10/4384" class="uri">http://www.pnas.org/content/90/10/4384</a></p>
<p>Stoddart JA., Taylor JF. 1988. Genotypic diversity: Estimation and prediction in samples. <em>Genetics</em> 118:705–711. Available at: <a href="http://www.genetics.org/content/118/4/705" class="uri">http://www.genetics.org/content/118/4/705</a></p>
</div>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
