<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="qc.html">Quality control</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/sandbox1/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="discriminant-analysis-of-principal-components-dapc" class="section level3">
<h3>Discriminant analysis of principal components (DAPC)</h3>
<ul>
<li>Reorganize DAPC plots</li>
<li>Explain cross-validation plots</li>
<li>Explain in detail what a DAPC plot show</li>
</ul>
<p>Another way of determining if the geographic populations are clustered is to use a discriminant analysis of principal components (DAPC). DAPC is a multivariate statistical approach that uses groups defined a priori to maximize among group variance and minimize within group variance. DAPC is explained in depth in the <a href="DAPC.html">DAPC chapter</a> on Part II of this tutorial or on the DAPC <a href="http://adegenet.r-forge.r-project.org/files%20/tutorial-dapc.pdf">adegenet vignette</a>.</p>
<p>The DAPC function is designed to work with genlight object <strong>with popluations defined a priori</strong>. For this excercise we will use the genlight object with all the SNP’s.</p>
<p>First, we will remove the <em>P. mirabilis</em> sample to allow us to focus on how the <em>P. infestans</em> samples group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1234</span>)
x &lt;-<span class="st"> </span><span class="kw">popsub</span>(x, <span class="dt">blacklist =</span> <span class="st">&quot;Pmir&quot;</span>)
dapc.x &lt;-<span class="st"> </span><span class="kw">dapc</span>(x, <span class="dt">pop =</span> <span class="kw">pop</span>(x), <span class="dt">n.pca=</span><span class="dv">10</span>, <span class="dt">n.da =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## Warning in dapc.genlight(x, pop = pop(x), n.pca = 10, n.da = 2): number of retained PCs of PCA may be too large (&gt; N /3)
##  results may be unstable</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">scatter</span>(dapc.x, <span class="dt">scree.da=</span><span class="ot">FALSE</span>,  <span class="dt">bg=</span><span class="st">&quot;white&quot;</span>, <span class="dt">pch=</span><span class="dv">17</span>:<span class="dv">22</span>, <span class="dt">cstar=</span><span class="dv">0</span>, <span class="dt">col=</span>cols)</code></pre></div>
<p><img src="genome_dapc_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">scatter</span>(dapc.x, <span class="dt">col =</span> cols, <span class="dt">cex =</span> <span class="dv">2</span>, <span class="dt">legend =</span> <span class="ot">TRUE</span>, <span class="dt">clabel =</span> F, <span class="dt">posi.leg =</span> <span class="st">&quot;bottomleft&quot;</span>, <span class="dt">scree.pca =</span> <span class="ot">TRUE</span>,
        <span class="dt">posi.pca =</span> <span class="st">&quot;topleft&quot;</span>, <span class="dt">cleg =</span> <span class="fl">0.75</span>)</code></pre></div>
<p><img src="genome_dapc_files/figure-html/unnamed-chunk-1-2.png" width="672" /></p>
<p>This figure doesn’t allow us to see any particular pattern, that why we use cross-validation. Very generally, cross-validation will objectivelly optimize the best trade-off point between retaining too many or too few PC’s for our model. The <a href="DAPC.html">DAPC chapter</a> on Part II of this tutorial explains why cross-validation is important. We need a genclone/genind object for this: the <code>x.gc.popsub</code> object. This object is a genclone object without the <em>P. mirabilis</em> sample:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(x)
grp &lt;-<span class="st"> </span><span class="kw">pop</span>(x)

xval &lt;-<span class="st"> </span><span class="kw">xvalDapc</span>(mat, grp, <span class="dt">n.rep =</span> <span class="dv">30</span>, <span class="dt">n.pca =</span> <span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">10</span>))</code></pre></div>
<pre><code>## Warning in xvalDapc(mat, grp, n.rep = 30, n.pca = c(1:10)): 1 group has
## only 1 member so it cannot be represented in both training and validation
## sets.</code></pre>
<p><img src="genome_dapc_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>The cross-validation shows that 5 PC is the ideal point of correctly predicted subsamples with the lowest error. we can plot the cross-validation DAPC results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">12345</span>)
<span class="kw">scatter</span>(xval$DAPC, <span class="dt">col =</span> cols, <span class="dt">cex =</span> <span class="dv">2</span>, <span class="dt">legend =</span> <span class="ot">TRUE</span>, <span class="dt">clabel =</span> <span class="ot">TRUE</span>, <span class="dt">posi.leg =</span> <span class="st">&quot;bottomleft&quot;</span>, <span class="dt">scree.pca =</span> <span class="ot">TRUE</span>,
        <span class="dt">posi.pca =</span> <span class="st">&quot;topleft&quot;</span>, <span class="dt">cleg =</span> <span class="fl">0.75</span>)</code></pre></div>
<p><img src="genome_dapc_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>The DAPC plot shows what we’ve found using all the other methods until now: There is no population aggrupation between the geographic locations of samples of <em>P. infestans</em></p>
<blockquote>
<p>Javier’s Opinion: This excercise is very long without all the other topics. We should discuss what we want for the workshop vs what we want for the book.</p>
</blockquote>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
