<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Introduction to R</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="qc.html">Quality control</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Introduction to R</h1>

</div>


<p>This appendix introduces basic use of the <a href="http://www.r-project.org">R</a> statistical and computer language and the <a href="http://www.rstudio.com/ide/">RStudio Desktop</a> integrated development environment (IDE). We assume you installed the needed resources mentioned in chapter 2 on getting ready to use R.</p>
<p><strong>Warning:</strong> To install R and RStudio you might need administrator privileges. Contact your IT manager and ask for an R installation or administrator privileges.</p>
<p>After installing both R and RStudio, please proceed to open RStudio. RStudio should look something like:</p>
<div class="figure">
<img src="images/Rstudio.png" alt="RStudio" />
<p class="caption">RStudio</p>
</div>
<p>The top left panel is a text-box where you can write a script. The bottom left panel is the console (or command line) where you directly execute commands and R writes results to the console; for example, type in <code>x &lt;- 1</code> hit return and then type <code>x</code> and hit return again and watch the console:</p>
<blockquote>
<p>Note: the “&gt;” symbol is R’s prompt for you to type something in the console.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>x &lt;-<span class="st"> </span><span class="dv">1</span>
&gt;<span class="st"> </span>x</code></pre></div>
<pre><code>[1] 1</code></pre>
<p>You just assigned the value 1 to the variable <code>x</code> with the assignment operator <code>&lt;-</code>. The second command asks R to show the value assigned to R. Finally, the top right panel shows variables loaded into the environment and the bottom right panel is useful for loading files, plotting graphs, installing packages and getting help. Feel free to explore all the various tabs or elements of each panel.</p>
<div id="assignment-data-types-and-operations" class="section level2">
<h2>Assignment, data types and operations</h2>
<p>Let’s learn a few simple facts about R:</p>
<p>R uses # to add comments to code:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span><span class="co"># Add 1 + 1 and write to console</span>
<span class="er">&gt;</span><span class="st"> </span><span class="dv">1</span> +<span class="st"> </span><span class="dv">1</span></code></pre></div>
<pre><code>[1] 2</code></pre>
<p>This is very useful when developing reusable code or when you want to remember a few months later what you actually did.</p>
<p>R is case sensitive so that <code>x &lt;- 5</code> is not the same as <code>X &lt;- x + 1</code>. Try typing this into your console and look at <code>x</code> and <code>X</code>, respectively:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>x &lt;-<span class="st"> </span><span class="dv">5</span>
&gt;<span class="st"> </span>X &lt;-<span class="st"> </span>x +<span class="st"> </span><span class="dv">1</span>
&gt;<span class="st"> </span>x</code></pre></div>
<pre><code>[1] 5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>X</code></pre></div>
<pre><code>[1] 6</code></pre>
<p>Next, let’s assign numbers to a vector:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>x &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>)
&gt;<span class="st"> </span>x</code></pre></div>
<pre><code>[1] 1 2 3 4 5</code></pre>
<p>You can see several important traits of R. The <code>c()</code> command (which stands for combine) can be used to make a list (e.g., a vector) of data. We can overwrite the variable <code>x</code> with new content and we can assign another data type (e.g., a single number becomes a vector). To compound things further you can even change the data type to strings overwriting the previous content:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>a &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Paris&quot;</span>, <span class="st">&quot;Tokyo&quot;</span>, <span class="st">&quot;Seattle&quot;</span>)
&gt;<span class="st"> </span>a</code></pre></div>
<pre><code>[1] &quot;Paris&quot;   &quot;Tokyo&quot;   &quot;Seattle&quot;</code></pre>
<p>Note that we use quotes to denote the string data type containing text.</p>
<p>To access individual elements of a vector, in this case the second element, we execute:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">&gt;<span class="st"> </span>a[<span class="dv">2</span>]</code></pre></div>
<pre><code>[1] &quot;Tokyo&quot;</code></pre>
</div>
<div id="creating-rstudio-projects" class="section level2">
<h2>Creating RStudio projects</h2>
<p>Coding projects usually results in a vast number of files, datasets, scripts and figures that can be difficult to manage and keep organized. RStudio has created the <strong>RStudio projects</strong> to facilitate and simplify the management a project by associating all of the project files into a working directory of your preference. This means that all script files, datasets and results of a project can be stored in a set location to facilitate access and management.</p>
<p>To create and RStudio project, we need to go to the upper right corner of RStudio, and click on <strong>New Project</strong>:</p>
<div class="figure">
<img src="images/RStudio_Proj1.png" />

</div>
<p>A pop-up window should appear. We want to create a new directory in our desktop to store our data and keep our files. To create the directory click on <strong>New Directory</strong>:</p>
<div class="figure">
<img src="images/RStudio_Proj2.png" />

</div>
<p>Since we are starting this project from scratch, we will select the <strong>Empty Project</strong> option. This will create a new RStudio project:</p>
<div class="figure">
<img src="images/RStudio_Proj3.png" />

</div>
<p>Then, we will use <em>APS_Workshop</em> as the name for our new directory, and we will create the directory as a subdirectory of your desktop:</p>
<div class="figure">
<img src="images/RStudio_Proj4.png" />

</div>
<p>And there is your new RStudio project withing the <code>APS_Workshop</code> directory! We can start downloading the required files and scripts for the workshop into that folder.</p>
</div>
<div id="console-tricks-code-completion-command-history" class="section level2">
<h2>Console tricks: Code completion &amp; command history</h2>
<p>RStudio has a very useful feature called code completion using the <kbd>Tab</kbd> key which can complete the full name of an object. For example type <code>hel</code> and hit <kbd>Tab</kbd> and you will see several functions pop up and you can select <code>help()</code>.</p>
<p>This also works inside a function to find function arguments. Type <code>help(</code> and hit <kbd>Tab</kbd> to select arguments for the help function.</p>
<p>RStudio records your command history. Thus you can scroll up or down the history of executed commands using the <em>Up</em> or <em>Down</em> arrows. Make sure your cursor is in the console and try to re-execute previous commands.</p>
<p>To quit R you can either use the <kbd>RStudio &gt; Quit</kbd> pull-down menu command or execute <kbd>⌘ + Q</kbd> (OS X) or <kbd>ctrl + Q</kbd> (PC).</p>
</div>
<div id="getting-help" class="section level2">
<h2>Getting help</h2>
<p>One way that R shines above other languages for analysis is the fact that R packages in CRAN are all documented. Help files are written in HTML and give the user a brief overview of:</p>
<ul>
<li>The purpose of a function</li>
<li>The parameters it takes</li>
<li>The output it yields</li>
<li>Some examples demonstrating its usage.</li>
</ul>
<p>To see all of the help topics in a package, you can simply type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help</span>(<span class="dt">package =</span> <span class="st">&quot;poppr&quot;</span>) <span class="co"># Get help for a package.</span>
<span class="kw">help</span>(amova)             <span class="co"># Get help for the amova function.</span>
?amova                  <span class="co"># same as above.</span>
??multilocus            <span class="co"># Search for functions that have the keyword multilocus.</span></code></pre></div>
<p>Some packages include vignettes that can have different formats such as being introductions, tutorials, or reference cards in PDF format. You can look at a list of vignettes in all packages by typing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>()                  <span class="co"># see vignettes from all packages</span>
<span class="kw">browseVignettes</span>(<span class="dt">package =</span> <span class="st">&#39;poppr&#39;</span>) <span class="co"># see vignettes from a specific package.</span></code></pre></div>
<p>and to look at a specific vignette you can type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">vignette</span>(<span class="st">&#39;poppr_manual&#39;</span>)</code></pre></div>
</div>
<div id="useful-resources-to-learn-r" class="section level2">
<h2>Useful resources to learn R</h2>
<p><em>Introductory</em></p>
<ul>
<li><a href="http://swirlstats.com/">Swirl</a> is a very well thought out R package that teaches you interactively.</li>
<li><a href="https://www.codeschool.com/courses/try-r">Code School Try R</a> is a nice interactive tutorial.</li>
<li><a href="http://www.statmethods.net/interface/help.html">Quick R</a></li>
<li><a href="http://cran.r-project.org/doc/contrib/Short-refcard.pdf">R reference card</a></li>
<li>A very nice, short <a href="http://ateucher.github.io/rcourse_site/index.html">introduction to R</a></li>
</ul>
<p><em>Advanced</em></p>
<ul>
<li><a href="http://adv-r.had.co.nz/">Advanced R</a> by Hadley Wickham</li>
</ul>
<p><em>Books</em>:</p>
<ul>
<li><a href="http://shop.oreilly.com/product/0636920022008.do">R in a Nutshell</a></li>
<li><a href="http://www.cookbook-r.com/">R cookbook</a> is a nice quick reference and tutorial for general R use.</li>
<li><a href="http://ggplot2.org/book/">ggplot2 book</a> is a useful reference if you want to customize graphs for publication.</li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
