<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Minimum spanning networks</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="quality_control.html">Quality control</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Minimum spanning networks</h1>
<h3 class="subtitle"><em><em>ZN Kamvar, SE Everhart and NJ Grünwald</em></em></h3>

</div>


<p>Minimum spanning networks (MSN) are a great way to visualize relationships among individuals in your data. Particularly for clonal organisms it can be a more powerful visualization tool than trees. In this chapter, we will show you how to construct and view minimum spanning networks on the command line and in an interactive viewer.</p>
<div id="from-the-command-line" class="section level1">
<h1>From the command line</h1>
<div id="note" class="section level3">
<h3>Note:</h3>
<p>This section will utilize Bruvo’s distance (Bruvo et al., 2004). Calculating Bruvo’s distance is computationally different to other distances; hence, this requires specialized functions for minimum spanning networks and bootstrapping that do not require a distance matrix:</p>
<table>
<thead>
<tr class="header">
<th>General function</th>
<th>Bruvo specific</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>poppr.msn</code></td>
<td><code>bruvo.msn</code></td>
</tr>
<tr class="even">
<td><code>aboot</code></td>
<td><code>bruvo.boot</code></td>
</tr>
</tbody>
</table>
</div>
<div id="minimum-spanning-network" class="section level3">
<h3>Minimum spanning network</h3>
<p>For this section, we will use the <code>monpop</code> data set from <span class="citation">(Everhart &amp; Scherm, 2015)</span>. See <a href="Genotypic_EvenRichDiv.html">Chapter 5</a> for more details. We will be focusing on sources of multilocus genotypes. The seasonal epidemic of the pathogen <em>Monilinia fructicola</em> begins with an ascospore (sexual propagule) released from a mummified peach fruit that had overwintered on the ground. It infects an emerging blossom that, in turn, asexually infects fruit, which proceed with cyclical, asexual infections. Two obvious questions are:</p>
<ol style="list-style-type: decimal">
<li>Are the major genotypes of Fruit Rot (FR) samples closely related?</li>
<li>To what degree do the Blossom Blight (BB) samples contribute to the FR?</li>
</ol>
<p>Let’s load the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;magrittr&quot;</span>)
<span class="kw">data</span>(monpop)
<span class="kw">splitStrata</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree<span class="op">/</span>Year<span class="op">/</span>Symptom
<span class="kw">summary</span>(monpop)</code></pre></div>
<pre><code>## 
## // Number of individuals: 694
## // Group sizes: 23 41 132 73 5 13 1 64 85 130 30 97
## // Number of alleles per locus: 3 6 3 11 9 5 8 7 5 9 8 11 10
## // Number of alleles per group: 48 53 58 48 37 40 13 44 60 64 60 63
## // Percentage of missing data: 0.51 %
## // Observed heterozygosity: 0</code></pre>
<p>We notice that tree number 26 is the only one to have been sampled for all three years. Let’s use it as an example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">t26 &lt;-<span class="st"> </span>monpop <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">setPop</span>(<span class="op">~</span>Tree) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">popsub</span>(<span class="st">&quot;26&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">setPop</span>(<span class="op">~</span>Year<span class="op">/</span>Symptom)
t26</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    155 multilocus genotypes 
##    390 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##      6 populations defined - 9_BB, 9_FR, 10_BB, 10_FR, 11_BB, 11_FR</code></pre>
<p>Now that we have our tree, let’s calculate a MSN using Bruvo’s distance <span class="citation">(Bruvo et al., 2004)</span>. Remember that this distance is based on a stepwise mutation model, so we have to first specify what kind of repeats units we have in our data (eg. dinucleotide = 2, trinucleotide = 3, etc.):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Set up our repeat lengths and populations to analyze</span>
reps &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">CHMFc4 =</span> <span class="dv">7</span>, <span class="dt">CHMFc5 =</span> <span class="dv">2</span>, <span class="dt">CHMFc12 =</span> <span class="dv">4</span>,
          <span class="dt">SEA =</span> <span class="dv">4</span>, <span class="dt">SED =</span> <span class="dv">4</span>, <span class="dt">SEE =</span> <span class="dv">2</span>, <span class="dt">SEG =</span> <span class="dv">6</span>,
          <span class="dt">SEI =</span> <span class="dv">3</span>, <span class="dt">SEL =</span> <span class="dv">4</span>, <span class="dt">SEN =</span> <span class="dv">2</span>,
          <span class="dt">SEP =</span> <span class="dv">4</span>, <span class="dt">SEQ =</span> <span class="dv">2</span>, <span class="dt">SER =</span> <span class="dv">4</span>)

sub9 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;9_BB&quot;</span>, <span class="st">&quot;9_FR&quot;</span>)

<span class="co"># Calculate the MSN</span>
t26.9msn &lt;-<span class="st"> </span><span class="kw">bruvo.msn</span>(t26, <span class="dt">replen =</span> reps, <span class="dt">sublist =</span> sub9, <span class="dt">showplot =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>The minimum spanning network is calculated via <code>bruvo.msn</code>. We have set the argument <code>showplot = FALSE</code> because we want to use the more powerful function <code>plot_poppr_msn</code> to view the MSN. I am telling it to label none of the samples, color populations using the “cm.colors” palette and scale the size of the nodes to <span class="math inline">\(log_{1.25}\)</span>. If you want to know what other things this function can do, simply type <code>help(&quot;plot_poppr_msn&quot;)</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Visualize the network</span>
<span class="kw">set.seed</span>(<span class="dv">120</span>)
<span class="kw">plot_poppr_msn</span>(t26, t26.9msn, <span class="dt">inds =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">palette =</span> cm.colors, <span class="dt">nodebase =</span> <span class="fl">1.25</span>)</code></pre></div>
<p><img src="Minimum_Spanning_Networks_files/figure-html/visualize_msn-1.png" width="700px" /></p>
<p>We can see that the Blossom Blight in the tree (pink pie slices) heavily contributed to the major groups of MLGs found in the Fruit Rot (blue pie pieces).</p>
<blockquote>
<p><strong>Try it for yourself!</strong> See if you can produce similar graphs with the 2010 and 2011 populations.</p>
</blockquote>
</div>
</div>
<div id="interactive-viewer" class="section level1">
<h1>Interactive viewer</h1>
<p>Creating a customized, publishable MSN is somewhat daunting as the <code>plot_poppr_msn</code> function has a whole lot arguments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">args</span>(plot_poppr_msn)</code></pre></div>
<pre><code>## function (x, poppr_msn, gscale = TRUE, gadj = 3, mlg.compute = &quot;original&quot;, 
##     glim = c(0, 0.8), gweight = 1, wscale = TRUE, nodebase = 1.15, 
##     nodelab = 2, inds = &quot;ALL&quot;, mlg = FALSE, quantiles = TRUE, 
##     cutoff = NULL, palette = NULL, layfun = layout.auto, beforecut = FALSE, 
##     pop.leg = TRUE, scale.leg = TRUE, ...) 
## NULL</code></pre>
<p>We created the interactive tool <code>imsn()</code> to facilitate interactive plotting. This function will be able to create minimum spanning networks from all genind objects in your current R session. In this section, we will recreate the plot above. First we will make sure that we have all the data we need and then we will run <code>imsn()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ls</span>() <span class="co"># Show all the data we have in our workspace</span></code></pre></div>
<pre><code>##  [1] &quot;Aeutamova&quot;           &quot;Aeutamovacc&quot;         &quot;Aeutccsignif&quot;       
##  [4] &quot;Aeut.new&quot;            &quot;Aeut.new.amova&quot;      &quot;Aeut.new.amova.test&quot;
##  [7] &quot;Aeutsignif&quot;          &quot;alpha&quot;               &quot;bib&quot;                
## [10] &quot;chrom&quot;               &quot;cols&quot;                &quot;contrib&quot;            
## [13] &quot;dapc.H3N2&quot;           &quot;dapc.results&quot;        &quot;dapc.x&quot;             
## [16] &quot;dna&quot;                 &quot;dna_file&quot;            &quot;dp&quot;                 
## [19] &quot;dp2&quot;                 &quot;dpf&quot;                 &quot;freq399&quot;            
## [22] &quot;freq906&quot;             &quot;gac&quot;                 &quot;gff&quot;                
## [25] &quot;gff_file&quot;            &quot;gi.rubi&quot;             &quot;gl.rubi&quot;            
## [28] &quot;grp&quot;                 &quot;gt&quot;                  &quot;H3N2&quot;               
## [31] &quot;html&quot;                &quot;i&quot;                   &quot;iPinf&quot;              
## [34] &quot;lambda&quot;              &quot;mat&quot;                 &quot;min_sample&quot;         
## [37] &quot;miss&quot;                &quot;monpop&quot;              &quot;monpop_diversity&quot;   
## [40] &quot;mon.tab&quot;             &quot;MX&quot;                  &quot;MXia&quot;               
## [43] &quot;mxpair&quot;              &quot;myIndex&quot;             &quot;myLevels&quot;           
## [46] &quot;myList&quot;              &quot;myPlots&quot;             &quot;myRegex&quot;            
## [49] &quot;myRows&quot;              &quot;N&quot;                   &quot;nan1&quot;               
## [52] &quot;nancycats&quot;           &quot;nanhwe.full&quot;         &quot;nanhwe.mat&quot;         
## [55] &quot;nanhwe.pop&quot;          &quot;nanmean&quot;             &quot;nanzero&quot;            
## [58] &quot;newmat&quot;              &quot;node.size&quot;           &quot;p&quot;                  
## [61] &quot;Pinf&quot;                &quot;pinflt&quot;              &quot;Pinf.ploidy&quot;        
## [64] &quot;plotrange&quot;           &quot;pnw.dapc&quot;            &quot;pop.data&quot;           
## [67] &quot;Pram&quot;                &quot;pramx&quot;               &quot;P.tab&quot;              
## [70] &quot;quants&quot;              &quot;reps&quot;                &quot;rubi.dist&quot;          
## [73] &quot;rubi.msn&quot;            &quot;rubi.pca&quot;            &quot;rubi.pca.scores&quot;    
## [76] &quot;rubi.VCF&quot;            &quot;SA&quot;                  &quot;SAia&quot;               
## [79] &quot;samps_per_row&quot;       &quot;sapair&quot;              &quot;snp399&quot;             
## [82] &quot;snp906&quot;              &quot;sub9&quot;                &quot;t26&quot;                
## [85] &quot;t26.9msn&quot;            &quot;temp&quot;                &quot;tree&quot;               
## [88] &quot;vcf&quot;                 &quot;vcf_file&quot;            &quot;x&quot;                  
## [91] &quot;X&quot;                   &quot;x.dist&quot;              &quot;x.msn&quot;              
## [94] &quot;x.msn2&quot;              &quot;x.tree&quot;              &quot;xval&quot;</code></pre>
<blockquote>
<p>Note: you will not be able to access your R console after running this function until you close the pop-up app.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">imsn</span>()</code></pre></div>
<p>(Note: you probably will not see the whole screen like this, but you can still scroll down)</p>
<div class="figure">
<img src="images/imsn_initial.png" alt="Intial view of imsn" />
<p class="caption">Intial view of imsn</p>
</div>
<p>We can see that there are five tabs and a sidebar. For now, we’ll explain what is on the sidebar.</p>
<div class="figure">
<img src="images/imsn_status.png" alt="status view" />
<p class="caption">status view</p>
</div>
<p>The first thing you see is a green bar that says “ready”. This means that there are no processes going on in the background and that it’s waiting for input. Below that are three buttons: <strong>Go!</strong>, <strong>reData</strong>, and <strong>reGraph</strong>. These tell the program that you are ready render your graphs.</p>
<div id="parameters-sidebar" class="section level2">
<h2>Parameters (sidebar)</h2>
<p>Notice that there are two main sections, <strong>Data Parameters</strong> and <strong>Graphical Parameters</strong>.</p>
<div id="data-parameters" class="section level3">
<h3>Data Parameters</h3>
<p>As you can expect, these options represent parameters that will manipulate your data. Changing any of these choices means that the distance matrix will be recalculated. When you modify things in this section and are satisfied with your choices, you can use the <strong>reData</strong> button to update the graph. You see here four choices presented:</p>
<ol style="list-style-type: decimal">
<li><strong>Choose dataset</strong> - This dropdown menu allows you to choose your data set.<br />
<img src="images/imsn_data_choose.png" alt="choose dataset" /></li>
<li><strong>Choose populations</strong> - this will change based on your data set, but it allows you to select different populations with which to subset your data<br />
<img src="images/imsn_population_choose.png" alt="choose populations" /></li>
<li><strong>Convert to genclone?</strong> - If your data is in genind format, you can convert it to genclone (minimally reduces time for clonal organisms)<br />
<img src="images/imsn_convert.png" alt="convert to genclone?" /></li>
<li><strong>Choose distance calculation</strong> - The distance measure to calculate from your data (notice that there’s “Custom” at the bottom)<br />
<img src="images/imsn_distance_choose.png" alt="choose distance calculation" /></li>
<li><strong>Distance arguments</strong> - every distance has different arguments, this is where you can modify them<br />
<img src="images/imsn_distance_args.png" alt="Distance arguments" /></li>
<li><strong>Include reticulations?</strong> - This is an important parameter that will include reticulations in the minimum spanning network. We’ll explore this with the <code>Pram</code> data set later.<br />
<img src="images/imsn_reticulate.png" alt="include reticulations?" /></li>
</ol>
</div>
<div id="graphical-parameters" class="section level3">
<h3>Graphical Parameters</h3>
<p>The parameters listed here allow you to modify the look and feel of the graph without having to recalculate the data. We encourage the user to play around with these to see what the different sliders and buttons do.</p>
</div>
</div>
<div id="output-tabs" class="section level2">
<h2>Output (tabs)</h2>
<p>Next let’s explore the different tabs on the MSN popup window:</p>
<div id="plot" class="section level3">
<h3>Plot</h3>
<p>This is the tab you see immediately. It has nothing in it because we haven’t told <code>imsn()</code> to do anything. Let’s hit <strong>Go!</strong> with our default parameters and see what happens.</p>
<div class="figure">
<img src="images/imsn_go.png" alt="minimum spanning network of partial_clone" />
<p class="caption">minimum spanning network of partial_clone</p>
</div>
<p>We now have a minimum spanning network of partial clone based a dissimilarity distance (“diss.dist”) where the distance is represented as counts of dissimilar alleles.</p>
</div>
<div id="data" class="section level3">
<h3>Data</h3>
<div class="figure">
<img src="images/imsn_data.png" alt="partial_clone output" />
<p class="caption">partial_clone output</p>
</div>
<p>This is the output of your data set. You can use this to confirm that your data is what you expect.</p>
</div>
<div id="command" class="section level3">
<h3>Command</h3>
<div class="figure">
<img src="images/imsn_command.png" alt="partial_clone command" />
<p class="caption">partial_clone command</p>
</div>
<p>If you copy and paste this command into your R console, you can recreate the minimum spanning network. This is important to make sure that your figure is reproducible.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;partial_clone&quot;</span>) <span class="co"># Don&#39;t forget to load the data</span>
partial_clone_sub &lt;-<span class="st"> </span><span class="kw">popsub</span>(partial_clone, <span class="dt">blacklist =</span> <span class="kw">character</span>(<span class="dv">0</span>))
partial_clone_dist &lt;-<span class="st"> </span><span class="kw">diss.dist</span>(partial_clone_sub, <span class="dt">percent =</span> <span class="ot">FALSE</span>, <span class="dt">mat =</span> <span class="ot">FALSE</span>)
min_span_net &lt;-<span class="st"> </span><span class="kw">poppr.msn</span>(partial_clone_sub, partial_clone_dist, <span class="dt">showplot =</span> <span class="ot">FALSE</span>, <span class="dt">include.ties =</span> <span class="ot">TRUE</span>)

<span class="kw">set.seed</span>(<span class="dv">69</span>)
<span class="kw">plot_poppr_msn</span>(partial_clone,
           min_span_net,
           <span class="dt">inds =</span> <span class="st">&quot;ALL&quot;</span>,
           <span class="dt">mlg =</span> <span class="ot">FALSE</span>,
           <span class="dt">gadj =</span> <span class="dv">3</span>,
           <span class="dt">nodebase =</span> <span class="fl">1.15</span>,
           <span class="dt">palette =</span> rainbow,
           <span class="dt">cutoff =</span> <span class="ot">NULL</span>,
           <span class="dt">quantiles =</span> <span class="ot">FALSE</span>,
           <span class="dt">beforecut =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Minimum_Spanning_Networks_files/figure-html/partial_clone_reproduce-1.png" width="700px" /></p>
</div>
<div id="save-plot" class="section level3">
<h3>Save Plot</h3>
<div class="figure">
<img src="images/imsn_save.png" />

</div>
<p>This allows you to save the plot to your computer in pdf or png format.</p>
</div>
<div id="session-information" class="section level3">
<h3>Session Information</h3>
<div class="figure">
<img src="images/imsn_sessioninfo.png" />

</div>
<p>This shows you what package versions were used to create the graph.</p>
</div>
</div>
<div id="re-creating-msns" class="section level2">
<h2>Re-creating MSNs</h2>
<p>We will now recreate the minimum spanning network that was displayed for the monpop data set.</p>
<p>First, choose the <code>t26</code> data set that we defined above:</p>
<div class="figure">
<img src="images/imsn_monpop_data.png" />

</div>
<p>Next, make sure you only select the “9_BB” and “9_FR” populations.</p>
<div class="figure">
<img src="images/imsn_monpop_pops.png" />

</div>
<p>Now, Select “Bruvo’s Distance”:</p>
<div class="figure">
<img src="images/imsn_monpop_bruvo.png" />

</div>
<p>Notice how the options have now changed. This is because bruvo’s distance is a special distance that is parametric. Don’t worry about the model for missing data, but in the SSR repeat lengths section, type “reps” (we defined this earlier):</p>
<div class="figure">
<img src="images/imsn_monpop_bruvo_opts.png" />

</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reps <span class="co"># our previous definition of &quot;reps&quot;</span></code></pre></div>
<pre><code>##  CHMFc4  CHMFc5 CHMFc12     SEA     SED     SEE     SEG     SEI     SEL 
##       7       2       4       4       4       2       6       3       4 
##     SEN     SEP     SEQ     SER 
##       2       4       2       4</code></pre>
<p>Now, modify the following graphical parameters:</p>
<ul>
<li>Node Size Scale: <strong>1.25</strong> (this is the option <code>nodebase</code>)</li>
<li>Random seed: <strong>120</strong></li>
<li>Labels: <strong>none</strong> (this is the option <code>inds</code>)</li>
<li>Palette: <strong>cm.colors</strong></li>
</ul>
<p>Once everthing is set, hit <strong>reData</strong> and the graph will reset just like this:</p>
<div class="figure">
<img src="images/imsn_monpop_plot.png" />

</div>
</div>
<div id="your-turn-use-the-data-for-phytophthora-ramorum" class="section level2">
<h2>Your turn (use the data for <em>Phytophthora ramorum</em>)</h2>
<p>Now we will use data from <span class="citation">Kamvar et al. (2015)</span>. This is the Sudden Oak Death pathogen collected from forests in Curry County, Oregon and nurseries in California and Oregon <span class="citation">(Goss et al., 2009)</span>. We will use this data set to see why reticulations are important for highly clonal organisms.</p>
<div id="load-the-data" class="section level3">
<h3>Load the data</h3>
<p>Exit the <code>imsn()</code> graphical window and execute the following statements:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;Pram&quot;</span>)
Pram</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##     98 multilocus genotypes 
##    729 diploid individuals
##      5 codominant loci
## 
## Population information:
## 
##      3 strata - SOURCE, YEAR, STATE
##      9 populations defined - 
## Nursery_CA, Nursery_OR, JHallCr_OR, ..., Winchuck_OR, ChetcoMain_OR, PistolRSF_OR</code></pre>
<p>This data has a few things in the “other” slot that we need. First is the repeat lengths called “REPLEN” and the other is a vector of hexadecimal colors for each population called “comparePal”.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">other</span>(Pram)<span class="op">$</span>REPLEN</code></pre></div>
<pre><code>##  PrMS6A1  Pr9C3A1 PrMS39A1 PrMS45A1 PrMS43A1 
##  3.00000  2.00000  4.00001  4.00000  4.00000</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">other</span>(Pram)<span class="op">$</span>comparePal</code></pre></div>
<pre><code>##    Nursery_CA    Nursery_OR    JHallCr_OR NFChetHigh_OR      Coast_OR 
##     &quot;#000000&quot;     &quot;#808080&quot;     &quot;#A6CEE3&quot;     &quot;#99CD91&quot;     &quot;#B89B74&quot; 
##   HunterCr_OR   Winchuck_OR ChetcoMain_OR  PistolRSF_OR 
##     &quot;#F06C45&quot;     &quot;#ED8F47&quot;     &quot;#825D99&quot;     &quot;#B15928&quot;</code></pre>
<p>We will use these to create the following minimum spanning network:</p>
<p><img src="Minimum_Spanning_Networks_files/figure-html/prammsn-1.png" width="700px" /></p>
<p>Here are the parameters you should use:</p>
<ul>
<li>Data:
<ul>
<li>Dataset: <strong>Pram</strong></li>
<li>Distance: <strong>Bruvo</strong></li>
<li>SSR Repeat Lengths: <strong><code>other(Pram)$REPLEN</code></strong> (Paste code into the pop-up app.)</li>
</ul></li>
<li>Graphical:
<ul>
<li>Grey Scale: <strong>9</strong></li>
<li>Node Size Scale: <strong>1.95</strong></li>
<li>Random Seed: <strong>78</strong></li>
<li>Labels: <strong>none</strong></li>
<li>Palette: <strong>Custom: <code>other(Pram)$comparePal</code></strong> (Paste code into the pop-up app.)</li>
</ul></li>
</ul>
<p>When you finish your session, you can exit the app by closing the window.</p>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-bruvo2004simple">
<p>Bruvo R., Michiels NK., D’Souza TG., Schulenburg H. 2004. A simple method for the calculation of microsatellite genotype distances irrespective of ploidy level. <em>Molecular Ecology</em> 13:2101–2106. Available at: <a href="http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x" class="uri">http://dx.doi.org/10.1111/j.1365-294X.2004.02209.x</a></p>
</div>
<div id="ref-everhart2014finescale">
<p>Everhart S., Scherm H. 2015. Fine-scale genetic structure of <em>Monilinia fructicola</em> during brown rot epidemics within individual peach tree canopies. <em>Phytopathology</em> 105:542–549. Available at: <a href="https://doi.org/10.1094/PHYTO-03-14-0088-R" class="uri">https://doi.org/10.1094/PHYTO-03-14-0088-R</a></p>
</div>
<div id="ref-goss2009population">
<p>Goss EM., Larsen M., Chastagner GA., Givens DR., Grünwald NJ. 2009. Population genetic analysis infers migration pathways of phytophthora ramorum in us nurseries. <em>PLoS Pathog</em> 5:e1000583. Available at: <a href="http://dx.doi.org/10.1371/journal.ppat.1000583" class="uri">http://dx.doi.org/10.1371/journal.ppat.1000583</a></p>
</div>
<div id="ref-kamvar2015spatial">
<p>Kamvar Z., Larsen M., Kanaskie A., Hansen E., Grünwald N. 2015. Spatial and temporal analysis of populations of the sudden oak death pathogen in oregon forests. <em>Phytopathology</em> 105:982–989. Available at: <a href="http://dx.doi.org/10.1094/PHYTO-12-14-0350-FI" class="uri">http://dx.doi.org/10.1094/PHYTO-12-14-0350-FI</a></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
