<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Population structure: G_{ST}, genetic distance, and clustering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="clustering_plot.html">Clustering plot</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="quality_control.html">Quality control</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="background_functions.html">Background_functions</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Population structure: <span class="math inline">\(G_{ST}\)</span>, genetic distance, and clustering</h1>
<h3 class="subtitle"><em><em>ZN Kamvar, SE Everhart and NJ Grünwald</em></em></h3>

</div>


<p>In this chapter we explore various ways of assessing if population are structured (e.g., differentiated). You can think of population structure as identifying clusters or groups of more closely related individuals resulting from reduced gene flow among these groups. Populations can be studied to determine if they are structured by using, for example, population differentiation summary statistics (e.g. <span class="math inline">\(G_{ST}\)</span>), clustering or <a href="Minimum_Spanning_Networks.html">minimum spanning networks</a>. Note, that this chapter will utilize many data sets due to the unique features offered by each. Let’s first look at an example of population differentiation based on <span class="math inline">\(G_{ST}\)</span>.</p>
<div id="g_st-an-example-with-felis-catus-data." class="section level2">
<h2><span class="math inline">\(G_{ST}\)</span> an example with <em>Felis catus</em> data.</h2>
<p>Assessing genetic diversity almost always starts with an analysis of a parameter such as <span class="math inline">\(G_{ST}\)</span>. There are <a href="http://www.molecularecologist.com/2011/03/should-i-use-fst-gst-or-d-2/">lengthy debates</a> as to what measure of differentiation is better <span class="citation">(Meirmans &amp; Hedrick, 2011)</span>. Instead of going into that lengthy debate, it would be more worthwhile to point you into the direction of a package dedicated to Modern Methods of Differentiation called <em>mmod</em>. We will use the data set <em>nancycats</em> containing 17 colonies of cats collected from Nancy, France. As cats tend to stay within small groups, we expect to see some population differentiation. In terms of these diversity measures, an index of <span class="math inline">\(G_{ST} = 0\)</span> indicates no differentiation, whereas <span class="math inline">\(G_{ST} = 1\)</span> indicates that populations are segregating for differing alleles.</p>
<p>Let’s load the package and the example data set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;mmod&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;nancycats&quot;</span>)
nancycats</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 237 individuals; 9 loci; 108 alleles; size: 145.3 Kb
## 
##  // Basic content
##    @tab:  237 x 108 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 8-18)
##    @loc.fac: locus factor for the 108 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: genind(tab = truenames(nancycats)$tab, pop = truenames(nancycats)$pop)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 9-23)
##    @other: a list containing: xy</code></pre>
<p>Now we will use Hendrick’s standardized <span class="math inline">\(G_{ST}\)</span> to assess population structure among these populations <span class="citation">(Hedrick, 2005)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Gst_Hedrick</span>(nancycats)</code></pre></div>
<pre><code>## $per.locus
##      fca8     fca23     fca43     fca45     fca77     fca78     fca90 
## 0.4750445 0.2956688 0.2675766 0.2653163 0.4855829 0.1933327 0.3807578 
##     fca96     fca37 
## 0.3913924 0.1609576 
## 
## $global
## [1] 0.3084895</code></pre>
<p>What does this output tell us?</p>
<p>Next we will look at genetic distance to find related groups of individuals.</p>
</div>
<div id="genetic-distance" class="section level2">
<h2>Genetic Distance</h2>
<p>If we wanted to analyze the relationship between individuals or populations, we would use genetic distance measures which calculate the “distance” between samples based on their genetic profile. These distances can be visualized with heatmaps, dendrograms, or minimum spanning networks. In the package <em>poppr</em>, there are several distances available:</p>
<table>
<thead>
<tr class="header">
<th>Distance</th>
<th>Function</th>
<th>Marker type</th>
<th>Can handle missing data</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Bruvo’s distance</td>
<td><code>bruvo.dist</code></td>
<td>microsatellite</td>
<td>yes</td>
<td></td>
</tr>
<tr class="even">
<td>Edwards’ distance</td>
<td><code>edwards.dist</code></td>
<td>any</td>
<td>no</td>
<td></td>
</tr>
<tr class="odd">
<td>Nei’s distance</td>
<td><code>nei.dist</code></td>
<td>any</td>
<td>no</td>
<td></td>
</tr>
<tr class="even">
<td>Provesti’s distance</td>
<td><code>provesti.dist</code></td>
<td>any</td>
<td>yes</td>
<td></td>
</tr>
<tr class="odd">
<td>Reynolds’ distance</td>
<td><code>reynolds.dist</code></td>
<td>any</td>
<td>no</td>
<td></td>
</tr>
<tr class="even">
<td>Rogers’ distance</td>
<td><code>rogers.dist</code></td>
<td>any</td>
<td>no</td>
<td></td>
</tr>
<tr class="odd">
<td>Provesti’s distance</td>
<td><code>bitwise.dist</code></td>
<td>SNP</td>
<td>yes</td>
<td></td>
</tr>
</tbody>
</table>
<p>One common way to visualize a genetic distance is with a dendrogram. For this example, we will use the <em>microbov</em> data set <span class="citation">(Laloe et al., 2007)</span>. This contains information on 704 cattle from both Africa and France over several different breeds. We can create a dendrogram over all 704 samples, but that would be difficult to visualize. For our purposes, let’s take ten random samples and calculate Provesti’s distance, which will return the fraction of the number of differences between samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ape&quot;</span>) <span class="co"># To visualize the tree using the &quot;nj&quot; function</span>
<span class="kw">library</span>(<span class="st">&quot;magrittr&quot;</span>)
<span class="kw">data</span>(microbov)
<span class="kw">set.seed</span>(<span class="dv">10</span>)
ten_samples &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">nInd</span>(microbov), <span class="dv">10</span>)
mic10       &lt;-<span class="st"> </span>microbov[ten_samples]
(micdist    &lt;-<span class="st"> </span><span class="kw">provesti.dist</span>(mic10))</code></pre></div>
<pre><code>##              FRBTBDA35243 AFBTSOM9386 FRBTBAZ26396 FRBTGAS9052 AFBIZEB9462
## AFBTSOM9386     0.7500000                                                 
## FRBTBAZ26396    0.6000000   0.6833333                                     
## FRBTGAS9052     0.6333333   0.8500000    0.5333333                        
## AFBIZEB9462     0.7166667   0.6666667    0.7833333   0.8500000            
## AFBTND211       0.6333333   0.5666667    0.6833333   0.7333333   0.8000000
## AFBTSOM9362     0.6000000   0.5333333    0.6666667   0.8166667   0.7000000
## AFBTSOM9360     0.7000000   0.5166667    0.6000000   0.7166667   0.7333333
## FRBTCHA25069    0.6166667   0.7000000    0.5833333   0.6833333   0.8000000
## FRBTBAZ26388    0.5333333   0.7000000    0.5500000   0.6333333   0.7500000
##              AFBTND211 AFBTSOM9362 AFBTSOM9360 FRBTCHA25069
## AFBTSOM9386                                                
## FRBTBAZ26396                                               
## FRBTGAS9052                                                
## AFBIZEB9462                                                
## AFBTND211                                                  
## AFBTSOM9362  0.5333333                                     
## AFBTSOM9360  0.5500000   0.5666667                         
## FRBTCHA25069 0.6500000   0.7333333   0.7166667             
## FRBTBAZ26388 0.6000000   0.6666667   0.6833333    0.6166667</code></pre>
<p>The above represents the pairwise distances between these 10 samples. We will use this distance matrix to create a neighbor-joining tree.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">theTree &lt;-<span class="st"> </span>micdist <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">nj</span>() <span class="op">%&gt;%</span><span class="st">    </span><span class="co"># calculate neighbor-joining tree</span>
<span class="st">  </span><span class="kw">ladderize</span>() <span class="co"># organize branches by clade</span>
<span class="kw">plot</span>(theTree)
<span class="kw">add.scale.bar</span>(<span class="dt">length =</span> <span class="fl">0.05</span>) <span class="co"># add a scale bar showing 5% difference.</span></code></pre></div>
<p><img src="Pop_Structure_files/figure-html/cattleplot-1.png" width="700px" /></p>
<p>Notice that the sample names start with either “AF” or “FR”. This indicates their country of origin and we are seeing that the populations cluster correspondingly. Of course, a tree is a hypothesis and one way of generating support is to bootstrap loci. This can be achieved with the <em>poppr</em> function <code>aboot</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">999</span>)
<span class="kw">aboot</span>(mic10, <span class="dt">dist =</span> provesti.dist, <span class="dt">sample =</span> <span class="dv">200</span>, <span class="dt">tree =</span> <span class="st">&quot;nj&quot;</span>, <span class="dt">cutoff =</span> <span class="dv">50</span>, <span class="dt">quiet =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Pop_Structure_files/figure-html/cattleboot-1.png" width="700px" /></p>
<pre><code>## 
## Phylogenetic tree with 10 tips and 8 internal nodes.
## 
## Tip labels:
##  FRBTBDA35243, AFBTSOM9386, FRBTBAZ26396, FRBTGAS9052, AFBIZEB9462, AFBTND211, ...
## Node labels:
##  100, NA, NA, 71.5, NA, NA, ...
## 
## Unrooted; includes branch lengths.</code></pre>
<p>The bootstrap value of 100 on the node separating the French and African samples gives support that the country of origin is a factor in how these breeds are structured. If we wanted to analyze all of the breeds against one another, it would be better to create a bootstrapped dendrogram based on a genetic distance. To do this, we will add 3 stratifications to the microbov data set: Country, Breed, and Species. We will then set the population to Country by Breed, convert the data to a genpop object and then create a tree using <code>aboot</code> with Nei’s genetic distance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Setting up the data</span>
<span class="kw">strata</span>(microbov) &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">other</span>(microbov))
microbov</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 704 individuals; 30 loci; 373 alleles; size: 1.1 Mb
## 
##  // Basic content
##    @tab:  704 x 373 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 5-22)
##    @loc.fac: locus factor for the 373 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: genind(tab = truenames(microbov)$tab, pop = truenames(microbov)$pop)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 30-61)
##    @strata: a data frame with 3 columns ( coun, breed, spe )
##    @other: a list containing: coun  breed  spe</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nameStrata</span>(microbov) &lt;-<span class="st"> </span><span class="er">~</span>Country<span class="op">/</span>Breed<span class="op">/</span>Species

<span class="co"># Analysis</span>
<span class="kw">set.seed</span>(<span class="dv">999</span>)
microbov <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">genind2genpop</span>(<span class="dt">pop =</span> <span class="op">~</span>Country<span class="op">/</span>Breed) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">aboot</span>(<span class="dt">cutoff =</span> <span class="dv">50</span>, <span class="dt">quiet =</span> <span class="ot">TRUE</span>, <span class="dt">sample =</span> <span class="dv">1000</span>, <span class="dt">distance =</span> nei.dist)</code></pre></div>
<pre><code>## 
##  Converting data from a genind to a genpop object... 
## 
## ...done.</code></pre>
<p><img src="Pop_Structure_files/figure-html/cattlepopboot-1.png" width="700px" /></p>
<pre><code>## 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##  AF_Borgou, AF_Zebu, AF_Lagunaire, AF_NDama, AF_Somba, FR_Aubrac, ...
## Node labels:
##  100, 100, 99.8, 93.1, 92.9, 63.8, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>Now we can see that, in all 1,000 bootstrapped trees, the African and French samples were each in separate clades. Of course, dendrograms are only one type of analysis you can use genetic distances for. Below is a table describing some of the different analyses for which you can utilize genetic distance:</p>
<table style="width:29%;">
<colgroup>
<col width="6%" />
<col width="8%" />
<col width="6%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th>Analysis</th>
<th>Function</th>
<th>Package</th>
<th>Note</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Bootstrapped dendrograms</td>
<td><code>aboot</code></td>
<td><em>poppr</em></td>
<td></td>
</tr>
<tr class="even">
<td>Mantel Test</td>
<td><code>mantel.randtest</code></td>
<td><em>ade4</em></td>
<td>To be used with geographic distance matrix</td>
</tr>
<tr class="odd">
<td>Principle Coordinates Analysis</td>
<td><code>cmdscale</code></td>
<td><em>stats</em></td>
<td></td>
</tr>
<tr class="even">
<td><a href="DAPC.html">DAPC</a></td>
<td><code>dapc</code></td>
<td><em>adegenet</em></td>
<td>Convert to matrix with <code>as.matrix</code></td>
</tr>
<tr class="odd">
<td><a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a></td>
<td><code>poppr.msn</code></td>
<td><em>poppr</em></td>
<td>requires a distance matrix; cannot handle genpop</td>
</tr>
</tbody>
</table>
</div>
<div id="k-means-hierarchical-clustering" class="section level2">
<h2>K-means hierarchical clustering</h2>
<p>A recent study reported that the origin of the potato late blight pathogen <em>Phytophthora infestans</em> lies in Mexico as opposed to South America <span class="citation">(Goss et al., 2014)</span>. We saw in the previous chapter that South American populations showed signatures of clonal reproduction while Mexican populations showed no evidence rejecting the null hypothesis of random mating. In this section, we will use K-means clustering in combination with bootstrapped dendrograms to see how well this pattern holds up. Clonal populations should have short terminal branch lengths and should cluster according to those branches. Panmictic populations will show no clear pattern. Let’s look at the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;Pinf&quot;</span>)
Pinf</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    72 multilocus genotypes 
##    86 tetraploid individuals
##    11 codominant loci
## 
## Population information:
## 
##     2 strata - Continent, Country
##     2 populations defined - South America, North America</code></pre>
<p>First, we will perform a cluster analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">MX &lt;-<span class="st"> </span><span class="kw">popsub</span>(Pinf, <span class="st">&quot;North America&quot;</span>)
MXclust &lt;-<span class="st"> </span><span class="kw">find.clusters</span>(MX)</code></pre></div>
<div class="figure">
<img src="Pop_Structure_files/figure-html/MXPCA.png" alt="MX_PCA" />
<p class="caption">MX_PCA</p>
</div>
<pre><code>## Choose the number PCs to retain (&gt;=1):</code></pre>
<pre><code>&gt; 50</code></pre>
<p>PC stands for principal components, which are unit-less transformations of your data that explain the variance observed. For the purposes of <code>find.clusters</code>, we can keep as many as we want.</p>
<div class="figure">
<img src="Pop_Structure_files/figure-html/MXCLUST.png" alt="MX_CLUSTER" />
<p class="caption">MX_CLUSTER</p>
</div>
<pre><code>## Choose the number PCs to retain (&gt;=2:</code></pre>
<pre><code>&gt; 3</code></pre>
<p>BIC stands for “Bayesian Information Criterion”. The lower the BIC value, the better. On the x axis are the number of clusters. We see that there is a bend at 3 clusters, indicating that the data clusters optimally into three groups.</p>
<p>And now we can see the cluster assignments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">MXclust</code></pre></div>
<pre><code>## $Kstat
## NULL
## 
## $stat
## NULL
## 
## $grp
##  PiMX01  PiMX02  PiMX03  PiMX04  PiMX05  PiMX06  PiMX07  PiMX10  PiMX11 
##       1       1       3       2       2       2       2       3       3 
##  PiMX12  PiMX13  PiMX14  PiMX15  PiMX16  PiMX17  PiMX18  PiMX19  PiMX20 
##       2       3       3       1       3       1       2       2       2 
##  PiMX21  PiMX22  PiMX23  PiMX24  PiMX25  PiMX26  PiMX27  PiMX28  PiMX29 
##       1       3       1       3       2       1       3       3       2 
##  PiMX30  PiMX40  PiMX41  PiMX42  PiMX43  PiMX44  PiMX45  PiMX46  PiMX47 
##       3       3       1       1       1       1       1       1       1 
##  PiMX48  PiMX49  PiMX50 PiMXT01 PiMXT02 PiMXT03 PiMXT04 PiMXT05 PiMXT06 
##       3       3       3       1       1       2       2       1       2 
## PiMXT07 PiMXt48 PiMXt68 
##       2       2       2 
## Levels: 1 2 3
## 
## $size
## [1] 17 16 15</code></pre>
<p>We will go through the same procedure for the South American population.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">SA &lt;-<span class="st"> </span><span class="kw">popsub</span>(Pinf, <span class="st">&quot;South America&quot;</span>)
SAclust &lt;-<span class="st"> </span><span class="kw">find.clusters</span>(SA)</code></pre></div>
<div class="figure">
<img src="Pop_Structure_files/figure-html/SAPCA.png" alt="SA_PCA" />
<p class="caption">SA_PCA</p>
</div>
<pre><code>## Choose the number PCs to retain (&gt;=1):</code></pre>
<pre><code>&gt; 30</code></pre>
<div class="figure">
<img src="Pop_Structure_files/figure-html/SACLUST.png" alt="SA_CLUSTER" />
<p class="caption">SA_CLUSTER</p>
</div>
<pre><code>## Choose the number PCs to retain (&gt;=2):</code></pre>
<pre><code>&gt; 4</code></pre>
<p>Notice here that there is no local minimum in the curve. This indicates that there <a href="http://lists.r-forge.r-project.org/pipermail/adegenet-forum/2011-June/000303.html">might not be enough information in the data set to properly cluster</a>. We will go ahead by choosing the highest number of clusters:</p>
<div id="trees" class="section level3">
<h3>Trees</h3>
<p>Now we will build trees. We are using Bruvo’s distance since polyploids bias calculation of other distances:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pinfreps &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">2</span>)
MXtree &lt;-<span class="st"> </span><span class="kw">bruvo.boot</span>(MX, <span class="dt">replen =</span> pinfreps, <span class="dt">cutoff =</span> <span class="dv">50</span>, <span class="dt">quiet =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Pop_Structure_files/figure-html/trees-1.png" width="700px" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">SAtree &lt;-<span class="st"> </span><span class="kw">bruvo.boot</span>(SA, <span class="dt">replen =</span> pinfreps, <span class="dt">cutoff =</span> <span class="dv">50</span>, <span class="dt">quiet =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Pop_Structure_files/figure-html/trees-2.png" width="700px" /></p>
<p>We see very long terminal branches in the MX tree. Let’s see how the groups we found with the clustering algorithm match up:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;ape&quot;</span>)
cols &lt;-<span class="st"> </span><span class="kw">rainbow</span>(<span class="dv">4</span>)
<span class="kw">plot.phylo</span>(MXtree, <span class="dt">cex =</span> <span class="fl">0.8</span>, <span class="dt">font =</span> <span class="dv">2</span>, <span class="dt">adj =</span> <span class="dv">0</span>, <span class="dt">tip.color =</span> cols[MXclust<span class="op">$</span>grp],
           <span class="dt">label.offset =</span> <span class="fl">0.0125</span>)
<span class="kw">nodelabels</span>(MXtree<span class="op">$</span>node.label, <span class="dt">adj =</span> <span class="kw">c</span>(<span class="fl">1.3</span>, <span class="op">-</span><span class="fl">0.5</span>), <span class="dt">frame =</span> <span class="st">&quot;n&quot;</span>, <span class="dt">cex =</span> <span class="fl">0.8</span>,
           <span class="dt">font =</span> <span class="dv">3</span>, <span class="dt">xpd =</span> <span class="ot">TRUE</span>)
<span class="kw">axisPhylo</span>(<span class="dv">3</span>)</code></pre></div>
<p><img src="Pop_Structure_files/figure-html/grouptree-1.png" width="700px" /></p>
<p>You can see that the assigned clusters don’t necessarily group with the dendrogram clusters. Let’s see what happens when we view this with the South American population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot.phylo</span>(SAtree, <span class="dt">cex =</span> <span class="fl">0.8</span>, <span class="dt">font =</span> <span class="dv">2</span>, <span class="dt">adj =</span> <span class="dv">0</span>, <span class="dt">tip.color =</span> cols[SAclust<span class="op">$</span>grp],
           <span class="dt">label.offset =</span> <span class="fl">0.0125</span>)
<span class="kw">nodelabels</span>(SAtree<span class="op">$</span>node.label, <span class="dt">adj =</span> <span class="kw">c</span>(<span class="fl">1.3</span>, <span class="op">-</span><span class="fl">0.5</span>), <span class="dt">frame =</span> <span class="st">&quot;n&quot;</span>, <span class="dt">cex =</span> <span class="fl">0.8</span>,
           <span class="dt">font =</span> <span class="dv">3</span>, <span class="dt">xpd =</span> <span class="ot">TRUE</span>)
<span class="kw">axisPhylo</span>(<span class="dv">3</span>)</code></pre></div>
<p><img src="Pop_Structure_files/figure-html/grouptreeSA-1.png" width="700px" /></p>
<p>Everything clusters together nicely, further supporting a non-panmictic population.</p>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-goss2014irish">
<p>Goss EM., Tabima JF., Cooke DEL., Restrepo S., Fry WE., Forbes GA., Fieland VJ., Cardenas M., Grünwald NJ. 2014. The Irish potato famine pathogen <em>Phytophthora infestans</em> originated in central mexico rather than the andes. <em>Proceedings of the National Academy of Sciences</em> 111:8791–8796. Available at: <a href="http://www.pnas.org/content/early/2014/05/29/1401884111.abstract" class="uri">http://www.pnas.org/content/early/2014/05/29/1401884111.abstract</a></p>
</div>
<div id="ref-hedrick2005standardized">
<p>Hedrick PW. 2005. A standardized genetic differentiation measure. <em>Evolution</em> 59:1633–1638. Available at: <a href="http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x" class="uri">http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x</a></p>
</div>
<div id="ref-laloe2007consensus">
<p>Laloe D., Jombart T., Dufour A-B., Moazami-Goudarzi K. 2007. Consensus genetic structuring and typological value of markers using multiple co-inertia analysis. <em>Genetics Selection Evolution</em> 39:545–567. Available at: <a href="http://dx.doi.org/10.1051/gse:2007021" class="uri">http://dx.doi.org/10.1051/gse:2007021</a></p>
</div>
<div id="ref-meirmans2011assessing">
<p>Meirmans PG., Hedrick PW. 2011. Assessing population structure: <span class="math inline">\(F_{ST}\)</span> and related measures. <em>Molecular Ecology Resources</em> 11:5–18. Available at: <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02927.x/full" class="uri">http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02927.x/full</a></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
