<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Populations strata: subsetting and clone correction</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="qc.html">Quality control</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/sandbox1/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Populations strata: subsetting and clone correction</h1>
<h3 class="subtitle"><em><em>ZN Kamvar, SE Everhart, and NJ Grünwald</em></em></h3>

</div>


<p>Populations are best sampled hierarchically on a range of scales from subpopulations (e.g. fields, valleys, ranges) to regions (e.g. valleys, states, countries or continents) or across time (years or decades). This approach is useful because population structure and evolutionary processes may not be discernible <em>a priori</em>. Most of the times we do not know if population are differentiated spatially or temporally. Thus, a combination of targeted local sampling with sampling over larger spatial or temporal scales is necessary to detect population structure over different scales, without using intense sampling throughout the entire range.</p>
<p>The methods implemented in <em>poppr</em> allow specification of which strata you want to analyze. This is a rapid way of working with subsets of your data without having to perform any data manipulation or changing the input file. In this tutorial, we will show you how to define the hierarchical structure of your data and how to specify specific levels that you might want to analyze.</p>
<div id="data-used-in-this-example" class="section level2">
<h2>Data used in this example</h2>
<p>For this example, we will use the <code>monpop</code> data set <span class="citation">(Everhart &amp; Scherm, 2015)</span>. This microsatellite data consists of 13 loci for 694 individuals of the haploid fungal pathogen <em>Monilinia fructicola</em> that infects peach flowers and fruits in commercial orchards. The monpop population came from four trees within a single orchard (trees 26, 45, and 7). Each tree was sampled in 2009, 2010, and/or 2011. Additionally, each sample was noted as to whether it came from a blossom or a fruit. This example data set is included with the <em>poppr</em> package.</p>
</div>
<div id="working-with-stratified-data" class="section level2">
<h2>Working with stratified data</h2>
<p>The steps for working with stratified data include:</p>
<ol style="list-style-type: decimal">
<li>Import data set with samples labeled according to strata</li>
<li>Define the stratifications for the data</li>
<li>Setting the stratification(s) that you want to have analyzed</li>
</ol>
</div>
<div id="importing-data-labeled-according-to-a-stratum" class="section level2">
<h2>Importing data labeled according to a stratum</h2>
<p>The easiest way to work with stratified data is to label each sample using an underscore “_” to separate each level. This was already done for the monpop data, where each sample was coded hierarchically by tree, year, and symptom in the following format: “Tree_Year_Symptom”.</p>
<p>Let’s load the hierarchically labeled example data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">data</span>(monpop)
monpop</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      1 stratum - Pop
##     12 populations defined - 
## 7_09_BB, 26_09_BB, 26_09_FR, ..., 45_10_FR, 26_11_BB, 26_11_FR</code></pre>
<p><strong>Genotype information</strong> shows us that the data contains 264 multilocus genotypes among 694 haploid individuals with 13 loci. <strong>Population information</strong> has two items, the stratifications and the populations defined. You can think of stratifications as the index names for each of the hierarchical levels within your data (so for our data it should be Tree, Year, and Symptom). By default, however, no stratifications are defined and so this is “Pop”, which is the entire dataset of 694 individuals. Because we labeled each sample according to stratification, populations defined shows us our data has 12 groups defined: 7_09_BB, 26_09_BB, 26_09_FR, 7_09_FR, 26_10_BB, 45_10_BB, 79_10_BB, 79_10_FR, 26_10_FR, 45_10_FR, 26_11_BB, and 26_11_FR.</p>
</div>
<div id="assigning-stratifications" class="section level2">
<h2>Assigning stratifications</h2>
<p>We imported the data that has three stratifications “Tree_Year_Symptom”. In order to analyze our data according to any combination of those three stratifications, we need to tell <em>poppr</em> that the 12 groups should be split by tree, year, and/or symptom. Thus, the first step is to split our data according to strata so that we can access each of the three hierarchical levels in the data. The <code>splitStrata</code> command is used to index the three stratifications:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">splitStrata</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree/Year/Symptom
monpop</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##     12 populations defined - 
## 7_09_BB, 26_09_BB, 26_09_FR, ..., 45_10_FR, 26_11_BB, 26_11_FR</code></pre>
<blockquote>
<p>The <code>splitStrata</code> command only needs to be run once: right after you import your data.</p>
</blockquote>
<p>After splitting the data populations are specified by stratification: “Tree Year Symptom”.</p>
<p>We can look at how the stratifications are distributed by using a <strong>treemap</strong> plot. This is a plot that allows us to visualize hierarchical stratifications. The function we’ll use is called <code>treemap()</code> from the <em>treemap</em> package.</p>
<p>The <code>treemap()</code> function needs only a data frame containing the strata and their respective counts. This can easily be done with the <em>dplyr</em> package where we will:</p>
<ol style="list-style-type: decimal">
<li>Group our strata by Tree, Year, and Symptom</li>
<li>Summarize the data by counting how many times we see each specific combination</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;dplyr&quot;</span>)
monstrata &lt;-<span class="st"> </span><span class="kw">strata</span>(monpop) %&gt;%<span class="st">     </span>
<span class="st">  </span><span class="kw">group_by</span>(Tree, Year, Symptom) %&gt;%
<span class="st">  </span><span class="kw">summarize</span>(<span class="dt">Count =</span> <span class="kw">n</span>())

monstrata</code></pre></div>
<pre><code>## Source: local data frame [12 x 4]
## Groups: Tree, Year [?]
## 
##      Tree   Year Symptom Count
##    &lt;fctr&gt; &lt;fctr&gt;  &lt;fctr&gt; &lt;int&gt;
## 1       7      9      BB    23
## 2       7      9      FR    73
## 3      26      9      BB    41
## 4      26      9      FR   132
## 5      26     10      BB     5
## 6      26     10      FR    85
## 7      26     11      BB    30
## 8      26     11      FR    97
## 9      45     10      BB    13
## 10     45     10      FR   130
## 11     79     10      BB     1
## 12     79     10      FR    64</code></pre>
<p>Now we can use the <code>treemap()</code> function to plot the data. Note that it has a lot of arguments to allow you to properly manipulate the graphic, but we will only use the very necessary components to visualize the distribution of the strata:</p>
<ul>
<li><strong>dtf</strong> - The data frame containing the stratifications and counts</li>
<li><strong>index</strong> - The variables used for nesting (in order)</li>
<li><strong>vSize</strong> - The variable to use to compute the size of the blocks</li>
</ul>
<p>All the other arguments we are using here give various aesthetics. If you want to know more about how they work, you can peruse the manual for the treemap function by typing <code>help(&quot;treemap&quot;, &quot;treemap&quot;)</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;treemap&quot;</span>)
monstrata          <span class="co"># Our data frame</span></code></pre></div>
<pre><code>## Source: local data frame [12 x 4]
## Groups: Tree, Year [?]
## 
##      Tree   Year Symptom Count
##    &lt;fctr&gt; &lt;fctr&gt;  &lt;fctr&gt; &lt;int&gt;
## 1       7      9      BB    23
## 2       7      9      FR    73
## 3      26      9      BB    41
## 4      26      9      FR   132
## 5      26     10      BB     5
## 6      26     10      FR    85
## 7      26     11      BB    30
## 8      26     11      FR    97
## 9      45     10      BB    13
## 10     45     10      FR   130
## 11     79     10      BB     1
## 12     79     10      FR    64</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nameStrata</span>(monpop) <span class="co"># The order of our variables</span></code></pre></div>
<pre><code>## [1] &quot;Tree&quot;    &quot;Year&quot;    &quot;Symptom&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">monstrata$Count    <span class="co"># The variable used for the block size</span></code></pre></div>
<pre><code>##  [1]  23  73  41 132   5  85  30  97  13 130   1  64</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Adjusting the aesthetics for the labels</span>
label_position &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&quot;center&quot;</span>, <span class="st">&quot;top&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;center&quot;</span>, <span class="st">&quot;center&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;center&quot;</span>, <span class="st">&quot;bottom&quot;</span>))
label_size     &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">Tree =</span> <span class="dv">0</span>, <span class="dt">Year =</span> <span class="dv">15</span>, <span class="dt">Symptom =</span> <span class="dv">15</span>)

<span class="co"># Plotting, First three arguments are necessary.</span>
<span class="kw">treemap</span>(<span class="dt">dtf =</span> monstrata, <span class="dt">index =</span> <span class="kw">nameStrata</span>(monpop), <span class="dt">vSize =</span> <span class="st">&quot;Count&quot;</span>,
        <span class="dt">type =</span> <span class="st">&quot;categorical&quot;</span>, <span class="dt">vColor =</span> <span class="st">&quot;Tree&quot;</span>, <span class="dt">title =</span> <span class="st">&quot;M. fructicola&quot;</span>,
        <span class="dt">align.labels =</span> label_position, <span class="dt">fontsize.labels =</span> label_size)</code></pre></div>
<p><img src="Population_Strata_files/figure-html/treemap-1.png" width="960" /></p>
<blockquote>
<p>You can also use the <code>itreemap()</code> function for an interactive exploration.</p>
</blockquote>
<p>Next, we analyze the data according to Tree and Year:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setPop</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree/Year
monpop</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##      6 populations defined - 7_9, 26_9, 26_10, 45_10, 79_10, 26_11</code></pre>
<p>To analyze the data according to Symptom:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setPop</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Symptom
monpop</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##      2 populations defined - BB, FR</code></pre>
<p>Order of the levels that you define is important, so if we wanted to define the symptoms according to tree, we would use the following:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setPop</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Symptom/Tree
monpop</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    694 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##      8 populations defined - BB_7, BB_26, FR_26, ..., BB_79, FR_79, FR_45</code></pre>
<p>Now that we have laid out the basics of manipulating data by strata, we will now apply strata for clone correction.</p>
</div>
<div id="clone-correction" class="section level2">
<h2>Clone correction</h2>
<p>When dealing with clonal populations, analyses are typically conducted with and without clone correction. Clone correction is a method of censoring a data set such that only one individual per MLG is represented per population <span class="citation">(Milgroom, 1996; Grünwald et al., 2003; Grünwald &amp; Hoheisel, 2006)</span>. This technique is commonly used with the index of association and genotypic diversity measures since clone corrected populations approximate behavior of sexual populations. Since we want to only observe unique genotypes per population, clone correction requires specification of the stratifications at which clones should be censored. This section will show how to clone correct at a specific stratification and also compare the results with uncorrected data.</p>
<blockquote>
<p>Question: Will allelic diversity increase or decrease with clone-censored data?</p>
</blockquote>
<p>Using <code>monpop</code> as an example, if we wanted to know the diversity of alleles within each tree per year, how should we go about correcting for the clones? We use the function <code>clonecorrect</code> specifying the “Tree/Year” strata:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mcc_TY &lt;-<span class="st"> </span><span class="kw">clonecorrect</span>(monpop, <span class="dt">strata =</span> ~Tree/Year, <span class="dt">keep =</span> <span class="dv">1</span>:<span class="dv">2</span>)
mcc_TY</code></pre></div>
<pre><code>## 
## This is a genclone object
## -------------------------
## Genotype information:
## 
##    264 multilocus genotypes 
##    278 haploid individuals
##     13 codominant loci
## 
## Population information:
## 
##      3 strata - Tree, Year, Symptom
##      6 populations defined - 7_9, 26_9, 26_10, 45_10, 79_10, 26_11</code></pre>
<p>Notice that the number of samples reduced from 694 to 278, but is still more than the number of MLGs. This indicates that there are duplicated genotypes that cross trees and years, but that’s okay because of our definition of a clone-corrected data set as having one representative genotype <em>per population</em>. Before we continue, we should set the original data to the same strata:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setPop</span>(monpop) &lt;-<span class="st"> </span><span class="er">~</span>Tree/Year</code></pre></div>
<p>Now we can compare the diversity of alleles at each locus using Simpson’s index (<span class="math inline">\(1-D\)</span>) as implemented in the function <code>locus_table</code> (Detailed in our <a href="Locus_Stats.html">chapter on locus based statistics</a>). We will do this in three steps:</p>
<ol style="list-style-type: decimal">
<li>Calculate diversity of the clone corrected data</li>
<li>Calculate diversity of the uncorrected data</li>
<li>Take the difference of <strong>step 1</strong> from <strong>step 2</strong>.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cc &lt;-<span class="st"> </span><span class="kw">locus_table</span>(mcc_TY, <span class="dt">info =</span> <span class="ot">FALSE</span>)
mp &lt;-<span class="st"> </span><span class="kw">locus_table</span>(monpop, <span class="dt">info =</span> <span class="ot">FALSE</span>)
mp -<span class="st"> </span>cc</code></pre></div>
<pre><code>##          summary
## locus     allele      1-D     Hexp Evenness
##   CHMFc4       .  0.00158  0.00031  0.00160
##   CHMFc5       . -0.09045 -0.09114 -0.04291
##   CHMFc12      .  0.00425  0.00312  0.03873
##   SEA          .  0.03122  0.02983  0.08509
##   SED          . -0.07392 -0.07564 -0.07563
##   SEE          . -0.04693 -0.04770 -0.02642
##   SEG          .  0.03202  0.03071  0.04818
##   SEI          . -0.00902 -0.01053  0.03060
##   SEL          . -0.00919 -0.01071 -0.05397
##   SEN          . -0.05006 -0.05176 -0.07643
##   SEP          .  0.01648  0.01511  0.04087
##   SEQ          . -0.05442 -0.05620 -0.08707
##   SER          . -0.04207 -0.04374 -0.02363
##   mean         . -0.02235 -0.02372 -0.01085</code></pre>
<p>Let’s plot <span class="math inline">\(1-D\)</span> so that we can see it better:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">locus_diff &lt;-<span class="st"> </span>mp -<span class="st"> </span>cc

<span class="co"># Note that I need to select the column containing Simpson&#39;s Index. That&#39;s</span>
<span class="co"># labeled as &quot;1-D&quot;.</span>
<span class="kw">barplot</span>(locus_diff[, <span class="st">&quot;1-D&quot;</span>], <span class="dt">ylab =</span> <span class="st">&quot;Change in Simpson&#39;s Index&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Locus&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;Comparison of clone-corrected vs. uncorrected data&quot;</span>)</code></pre></div>
<p><img src="Population_Strata_files/figure-html/simpdivplot-1.png" width="672" /></p>
<p>We can see quite a difference in some loci after clone correcting based on tree in the overall data set showing that, while some loci show a decrease, many loci show an increase in allelic diversity after clone-correction.</p>
<div id="advanced-r-writing-your-own-functions-for-analysis-by-stratum" class="section level3">
<h3>Advanced R: writing your own functions for analysis by stratum</h3>
<p>Of course, we still want to analyze each tree/year combination separately. Instead of typing those commands above for each combination, we can write a function to do it for us.</p>
<p>A function can be thought of as a set of instructions that tells the computer what to do. You have been using functions before such as <code>poppr()</code>. A function is written like this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fun_name &lt;-<span class="st"> </span>function(arguments){
  <span class="co"># instructions</span>
}</code></pre></div>
<p><code>fun_name</code> is the name of the function, the R command <code>function</code> tells R that you are about to write a function. Arguments are variables that you will pass to the function and the curly braces (<code>{</code> and <code>}</code>) provide the area where you write all the steps of the function. Next, we will go through an example of writing a function.</p>
<div id="example-harmonic-mean" class="section level4">
<h4>Example: harmonic mean</h4>
<p>Let’s write a function that will take the harmonic mean of a vector of numbers. First, we’ll see how to do this calculation step by step before we write a function.</p>
<blockquote>
<p>The harmonic mean has been utilized to estimate effective population size of a population that has undergone drastic changes in population size <span class="citation">(Nielsen &amp; Slatkin, 2013)</span>.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">psize &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">50</span>, <span class="dv">80</span>, <span class="dv">20</span>, <span class="dv">500</span>)
<span class="co"># Arithmetic mean</span>
<span class="kw">mean</span>(psize)</code></pre></div>
<pre><code>## [1] 126.6667</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Harmonic mean</span>
rmean &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="dv">1</span>/psize)
<span class="dv">1</span>/rmean</code></pre></div>
<pre><code>## [1] 30.84833</code></pre>
<p>Notice that there are two steps to calculating the harmonic mean. If we wanted to calculate the harmonic mean of a lot of different vectors, this would get really repetitive. A function is a way of encapsulating a lot of instructions to reduce the number of steps (and potential errors) you would need to take in your analysis. Below, we will write a function to calculate the harmonic mean using the steps above.</p>
<blockquote>
<p>Note here that the <em>argument</em> is <strong>x</strong> and it is a placeholder for any value you put into the function.</p>
</blockquote>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hmean &lt;-<span class="st"> </span>function(x){
  rmean &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="dv">1</span>/x)
  <span class="dv">1</span>/rmean
}
<span class="kw">mean</span>(psize)</code></pre></div>
<pre><code>## [1] 126.6667</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hmean</span>(psize)</code></pre></div>
<pre><code>## [1] 30.84833</code></pre>
</div>
<div id="writing-a-function-for-comparing-clone-corrected-and-uncorrected-data" class="section level4">
<h4>Writing a function for comparing clone corrected and uncorrected data</h4>
<p>We will simply take the steps from above and turn them into a function to calculate the difference in Simpson’s diversity for a given population. In order to do that, we will need to compute three steps:</p>
<ol style="list-style-type: decimal">
<li>the name of the population</li>
<li>the clone-corrected data set</li>
<li>the uncorrected data set</li>
</ol>
<p>From there, we can construct the function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plot_simp_diff &lt;-<span class="st"> </span>function(pop_name, clone_corrected, un_corrected){

  <span class="co"># Step 1: calculate diversity for clone-corrected data</span>
  cc &lt;-<span class="st"> </span><span class="kw">locus_table</span>(clone_corrected, <span class="dt">pop =</span> pop_name, <span class="dt">info =</span> <span class="ot">FALSE</span>)

  <span class="co"># Step 2: calculate diversity for uncorrected data</span>
  uc &lt;-<span class="st"> </span><span class="kw">locus_table</span>(un_corrected, <span class="dt">pop =</span> pop_name, <span class="dt">info =</span> <span class="ot">FALSE</span>)

  <span class="co"># Step 3: Take the difference</span>
  res &lt;-<span class="st"> </span>uc -<span class="st"> </span>cc

  <span class="co"># Step 4: Plot Simpson&#39;s index.</span>
  <span class="kw">barplot</span>(res[, <span class="st">&quot;1-D&quot;</span>], <span class="dt">main =</span> pop_name, <span class="dt">ylab =</span> <span class="st">&quot;Change in Simpson&#39;s Index&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;Locus&quot;</span>)
}</code></pre></div>
<p>Now that we have our function written, let’s gather the population names and construct a <em>for loop</em> that will analyze each one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)) <span class="co"># Set up the graphics to have two rows and three columns</span>

for (i in <span class="kw">popNames</span>(monpop)){
  <span class="kw">plot_simp_diff</span>(i, mcc_TY, monpop)
}</code></pre></div>
<p><img src="Population_Strata_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)) <span class="co"># Next we reset the graphics to have one row and one column</span></code></pre></div>
<p>These barplots show the difference in Simpson’s index of original minus clone corrected data for each population per locus. We can see that allelic diversity generally is lower in the total data set (containing some repeated MLGs) relative to clone corrected data.</p>
</div>
</div>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>This was a brief introduction to the easiest way to create stratifications and apply them in <em>poppr</em> to more rapidly analyze your data. By indexing the stratifications of your data, you can set the stratification(s) you want to have analyzed in a single command. This approach avoids having to create new sub-sets of the data for each analysis and simultaneously reduces the chance of error when manipulating data sets by hand.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<!-------->
<div id="refs" class="references">
<div id="ref-everhart2014finescale">
<p>Everhart S., Scherm H. 2015. Fine-scale genetic structure of <em>Monilinia fructicola</em> during brown rot epidemics within individual peach tree canopies. <em>Phytopathology</em> 105:542–549. Available at: <a href="https://doi.org/10.1094/PHYTO-03-14-0088-R" class="uri">https://doi.org/10.1094/PHYTO-03-14-0088-R</a></p>
</div>
<div id="ref-grunwald2003">
<p>Grünwald NJ., Goodwin SB., Milgroom MG., Fry WE. 2003. Analysis of genotypic diversity data for populations of microorganisms. <em>Phytopathology</em> 93:738–746. Available at: <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738" class="uri">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738</a></p>
</div>
<div id="ref-grunwald2006hierarchical">
<p>Grünwald NJ., Hoheisel G-A. 2006. Hierarchical analysis of diversity, selfing, and genetic differentiation in populations of the oomycete <em>Aphanomyces euteiches</em>. <em>Phytopathology</em> 96:1134–1141. Available at: <a href="http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134" class="uri">http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO-96-1134</a></p>
</div>
<div id="ref-milgroom1996recombination">
<p>Milgroom MG. 1996. Recombination and the multilocus structure of fungal populations. <em>Annual Review of Phytopathology</em> 34:457–477. Available at: <a href="http://www.annualreviews.org/doi/abs/10.1146/annurev.phyto.34.1.457" class="uri">http://www.annualreviews.org/doi/abs/10.1146/annurev.phyto.34.1.457</a></p>
</div>
<div id="ref-nielsen2013">
<p>Nielsen R., Slatkin M. 2013. <em>An introduction to population genetics: Theory and applications</em>. Sinauer Associates, Incorporated. Available at: <a href="http://books.google.com/books?id=Iy08kgEACAAJ" class="uri">http://books.google.com/books?id=Iy08kgEACAAJ</a></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
