<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="BJ Knaus and NJ Grünwald" />


<title>Quality control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="quality_control.html">Quality control</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="background_functions.html">Background_functions</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Quality control</h1>
<h4 class="author"><em>BJ Knaus and NJ Grünwald</em></h4>

</div>


<style type="text/css">
pre code, pre, code {
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
</style>
<p>An important issue to consider once you have your file of genotypes is that not all samples and variants are of the same quality. Some samples may not have been sequenced at a high coverage and may need to be considered failed samples or samples that require additional sequencing. Some variants may not have been sequenced at a sufficient depth and may need to be considered low quality. Some variants may be located in repetetive elements and result in unusually high coverage and may include more than two alleles (when working with a diploid organism). Most variant callers claim to “aggressively” call variants with the intention that they will see a quality control step. In this section we explore our data and develop strategies for quality control.</p>
<div id="data-input" class="section level2">
<h2>Data input</h2>
<p>One nice feature of CRAN and R is that it allows use to include data as packages. However, because it is part of a package it needs to be in a child directory of the package. Here we’ll use <code>system.file()</code> to help us find the file and then read it in. In your workflows you’ll typically know where your file is so you can simply use the file name.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vcfR)
<span class="kw">library</span>(pinfsc50)
myVcf &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;pinf_sc50.vcf.gz&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;pinfsc50&quot;</span>)
vcf &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">read.vcfR</span>(myVcf, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>If you set <code>verbose</code> to <code>TRUE</code> you will receive information on progress.</p>
</div>
<div id="show-method" class="section level2">
<h2>Show method</h2>
<p>The first step of examining the data should include verifying that you successfully read it in. We can use the <code>show</code> method to get a summary of the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">show</span>(vcf)</code></pre></div>
<pre class="r-output"><code>## ***** Object of Class vcfR *****
## 18 samples
## 1 CHROMs
## 22,031 variants
## Object size: 22.4 Mb
## 7.929 percent missing data
## *****        *****         *****
</code></pre>
<p>You should know the number of samples in your project and this should match the number reported by <code>show</code>. It is usually good to try to validate the number of variants read in matches the number in your file. We can accomplish this with the function <code>count.fields()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">count.fields</span>(myVcf))</code></pre></div>
<pre class="r-output"><code>## [1] 22031
</code></pre>
<p>This number should match the number reported by <code>show()</code>. If they do match, and the number of samples is correct, then we have validated that we have the correct number of rows and columns and can proceed.</p>
<blockquote>
<p>Bonus question: why does this only count variants and not meta lines?</p>
</blockquote>
</div>
<div id="the-meta-section" class="section level2">
<h2>The meta section</h2>
<p>Different VCF files created by different variant callers, or the same variant caller parameterized differently, may produce VCF files with different information in them. To get a first peak at what data we have we can use the <code>queryMETA()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">queryMETA</span>(vcf)</code></pre></div>
<pre class="r-output"><code>##  [1] "FILTER=ID=LowQual"                    "FORMAT=ID=AD"                         "FORMAT=ID=DP"                         "FORMAT=ID=GQ"                         "FORMAT=ID=GT"                        
##  [6] "FORMAT=ID=PL"                         "GATKCommandLine=ID=HaplotypeCaller"   "INFO=ID=AC"                           "INFO=ID=AF"                           "INFO=ID=AN"                          
## [11] "INFO=ID=BaseQRankSum"                 "INFO=ID=ClippingRankSum"              "INFO=ID=DP"                           "INFO=ID=DS"                           "INFO=ID=FS"                          
## [16] "INFO=ID=HaplotypeScore"               "INFO=ID=InbreedingCoeff"              "INFO=ID=MLEAC"                        "INFO=ID=MLEAF"                        "INFO=ID=MQ"                          
## [21] "INFO=ID=MQ0"                          "INFO=ID=MQRankSum"                    "INFO=ID=QD"                           "INFO=ID=ReadPosRankSum"               "INFO=ID=SOR"                         
## [26] "1 contig=&lt;IDs omitted from queryMETA"
</code></pre>
<p>The <code>meta</code> section is a place where acronyms are defined that are used elsewhere in the file. If we tell the function what element to choose we get more information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">queryMETA</span>(vcf, <span class="dt">element =</span> <span class="st">&quot;DP&quot;</span>)</code></pre></div>
<pre class="r-output"><code>## [[1]]
## [1] "FORMAT=ID=DP"                                                                          "Number=1"                                                                             
## [3] "Type=Integer"                                                                          "Description=Approximate read depth (reads with MQ=255 or with bad mates are filtered)"
## 
## [[2]]
## [1] "INFO=ID=DP"                                                            "Number=1"                                                             
## [3] "Type=Integer"                                                          "Description=Approximate read depth; some reads may have been filtered"
</code></pre>
<p>The <code>DP</code> acronym occurs in the <code>INFO</code> column (column eight) as well as in the <code>FORMAT</code> column (column nine) so it is defined twice.</p>
</div>
<div id="extracting-data-from-the-gt-section" class="section level2">
<h2>Extracting data from the gt section</h2>
<p>The genotypes are in a tabular format. The genotypes are typically accompanied with colon delimited data. Note that according to the <a href="http://samtools.github.io/hts-specs/">VCF specification</a> each row can have a different format, so each row needs to be processed separately.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">vcf<span class="op">@</span>gt[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</code></pre></div>
<pre class="r-output"><code>##      FORMAT           BL2009P4_us23              DDR7602                    IN2009T1_us22              LBUS5                      NL07434                    
## [1,] "GT:AD:DP:GQ:PL" "1|1:0,7:7:21:283,21,0"    "1|1:0,6:6:18:243,18,0"    "1|1:0,8:8:24:324,24,0"    "1|1:0,6:6:18:243,18,0"    "1|1:0,12:12:36:486,36,0"  
## [2,] "GT:AD:DP:GQ:PL" "0|0:12,0:12:36:0,36,427"  "0|0:20,0:20:60:0,60,819"  "0|0:16,0:16:48:0,48,650"  "0|0:20,0:20:60:0,60,819"  "0|0:28,0:28:84:0,84,948"  
## [3,] "GT:AD:DP:GQ:PL" "0|0:27,0:27:81:0,81,1117" "0|0:26,0:26:78:0,78,1077" "0|0:23,0:23:69:0,69,946"  "0|0:26,0:26:78:0,78,1077" "0|1:19,20:39:99:565,0,559"
## [4,] "GT:AD:DP:GQ:PL" "0|0:29,0:29:87:0,87,1243" "0|0:27,0:27:81:0,81,1158" "0|0:32,0:32:96:0,96,1299" "0|0:27,0:27:81:0,81,1158" "0|1:19,19:38:99:523,0,535"
</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dp &lt;-<span class="st"> </span><span class="kw">extract.gt</span>(vcf,  <span class="dt">element =</span> <span class="st">&quot;DP&quot;</span>, <span class="dt">as.numeric =</span> <span class="ot">TRUE</span>)
<span class="kw">class</span>(dp)</code></pre></div>
<pre class="r-output"><code>## [1] "matrix"
</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(dp)</code></pre></div>
<pre class="r-output"><code>## [1] 22031    18
</code></pre>
<p>We have now taken our VCF data, a format that most R functions can not work with, and converted part of it into a matrix, a common data structure that many R functions can work with. One way to visualize this data is with violin plots. Here we’ll use <code>ggplot2</code> to create violin plots of depth.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(reshape2)

dpf &lt;-<span class="st"> </span><span class="kw">melt</span>(dp, <span class="dt">varnames =</span> <span class="kw">c</span>(<span class="st">&quot;Index&quot;</span>, <span class="st">&quot;Sample&quot;</span>),
            <span class="dt">value.name =</span> <span class="st">&quot;Depth&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
dpf &lt;-<span class="st"> </span>dpf[ dpf<span class="op">$</span>Depth <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, ]
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dpf, <span class="kw">aes</span>(<span class="dt">x =</span> Sample, <span class="dt">y =</span> Depth))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_violin</span>(<span class="dt">fill =</span> <span class="st">&quot;#C0C0C0&quot;</span>, <span class="dt">adjust =</span> <span class="fl">1.0</span>,
                     <span class="dt">scale =</span> <span class="st">&quot;count&quot;</span>, <span class="dt">trim =</span> <span class="ot">TRUE</span>)
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),
               <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans =</span> scales<span class="op">::</span><span class="kw">log2_trans</span>(),
                            <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">800</span>),
                            <span class="dt">minor_breaks =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">10</span> <span class="op">*</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">8</span> <span class="op">*</span><span class="st"> </span><span class="dv">100</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.major.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#A9A9A9&quot;</span>, <span class="dt">size =</span> <span class="fl">0.6</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.minor.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#C0C0C0&quot;</span>, <span class="dt">size =</span> <span class="fl">0.2</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Depth (DP)&quot;</span>)
p</code></pre></div>
<p><img src="quality_control_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We see that the data are generally unimodal which is what we would expect from genomic data. The bulging at the bottom of the plot is because we have integer data (i.e., we have information for 1 and 2 but no fractional values in between). At least one sample, P1362, looks questionable in quality. While most samples were sequenced at around 20X, all samples also include long tails well into the hundres of times sequenced. You might consider this undesireable. You may also consider low coverage variants questionable as well We can manage this by omitting tham. Here we’ll the 10th and 90th percentile to identify variants of unusual depth. First we’ll omit them by marking them as missing dataa (<code>NA</code>). Then we’ll omit samples and variants with an unusually high amount of missing data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">quants &lt;-<span class="st"> </span><span class="kw">apply</span>(dp, <span class="dt">MARGIN =</span> <span class="dv">2</span>, quantile, <span class="dt">probs =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.9</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
dp2 &lt;-<span class="st"> </span><span class="kw">sweep</span>(dp, <span class="dt">MARGIN =</span> <span class="dv">2</span>, <span class="dt">FUN =</span> <span class="st">&quot;-&quot;</span>, quants[<span class="dv">1</span>, ])
dp[dp2 <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
dp2 &lt;-<span class="st"> </span><span class="kw">sweep</span>(dp, <span class="dt">MARGIN =</span> <span class="dv">2</span>, <span class="dt">FUN =</span> <span class="st">&quot;-&quot;</span>, quants[<span class="dv">2</span>, ])
dp[dp2 <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
dp[dp <span class="op">&lt;</span><span class="st"> </span><span class="dv">4</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
<span class="co"># Update the vcfR object with our changes.</span>
vcf<span class="op">@</span>gt[, <span class="op">-</span><span class="dv">1</span>][ <span class="kw">is.na</span>(dp) <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span> ] &lt;-<span class="st"> </span><span class="ot">NA</span>
vcf</code></pre></div>
<pre class="r-output"><code>## ***** Object of Class vcfR *****
## 18 samples
## 1 CHROMs
## 22,031 variants
## Object size: 20.5 Mb
## 27.7 percent missing data
## *****        *****         *****
</code></pre>
<p>Now we’ll omit samples that are over 55% missing data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dp &lt;-<span class="st"> </span><span class="kw">extract.gt</span>(vcf,  <span class="dt">element =</span> <span class="st">&quot;DP&quot;</span>, <span class="dt">as.numeric =</span> <span class="ot">TRUE</span>)
myMiss &lt;-<span class="st"> </span><span class="kw">apply</span>(dp, <span class="dt">MARGIN =</span> <span class="dv">2</span>, <span class="cf">function</span>(x){<span class="kw">sum</span>( <span class="kw">is.na</span>(x))})
myMiss &lt;-<span class="st"> </span>myMiss <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(dp)
vcf<span class="op">@</span>gt &lt;-<span class="st"> </span>vcf<span class="op">@</span>gt[, <span class="kw">c</span>(<span class="ot">TRUE</span>, myMiss <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.55</span>)]
vcf</code></pre></div>
<pre class="r-output"><code>## ***** Object of Class vcfR *****
## 17 samples
## 1 CHROMs
## 22,031 variants
## Object size: 20.2 Mb
## 26.1 percent missing data
## *****        *****         *****
</code></pre>
<p>Now we’ll omit variants that are more than 20% missing data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myMiss &lt;-<span class="st"> </span><span class="kw">apply</span>(dp, <span class="dt">MARGIN =</span> <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="kw">sum</span>(<span class="kw">is.na</span>(x))})
myMiss &lt;-<span class="st"> </span>myMiss <span class="op">/</span><span class="st"> </span><span class="kw">ncol</span>(dp)
vcf &lt;-<span class="st"> </span>vcf[myMiss <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.2</span>, ]
vcf</code></pre></div>
<pre class="r-output"><code>## ***** Object of Class vcfR *****
## 17 samples
## 1 CHROMs
## 11,715 variants
## Object size: 12.3 Mb
## 6.714 percent missing data
## *****        *****         *****
</code></pre>
<p>Once we’ve processed the data its good to visualize how those decisions have afffected the data. We’ll make another set of violin plots and compare them to the ones we made previously.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dp &lt;-<span class="st"> </span><span class="kw">extract.gt</span>(vcf,  <span class="dt">element =</span> <span class="st">&quot;DP&quot;</span>, <span class="dt">as.numeric =</span> <span class="ot">TRUE</span>)
dpf &lt;-<span class="st"> </span><span class="kw">melt</span>(dp, <span class="dt">varnames =</span> <span class="kw">c</span>(<span class="st">&quot;Index&quot;</span>, <span class="st">&quot;Sample&quot;</span>),
            <span class="dt">value.name =</span> <span class="st">&quot;Depth&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
dpf &lt;-<span class="st"> </span>dpf[ dpf<span class="op">$</span>Depth <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, ]
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dpf, <span class="kw">aes</span>(<span class="dt">x =</span> Sample, <span class="dt">y =</span> Depth))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_violin</span>(<span class="dt">fill =</span> <span class="st">&quot;#C0C0C0&quot;</span>, <span class="dt">adjust =</span> <span class="fl">1.0</span>,
                     <span class="dt">scale =</span> <span class="st">&quot;count&quot;</span>, <span class="dt">trim =</span> <span class="ot">TRUE</span>)
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),
               <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans =</span> scales<span class="op">::</span><span class="kw">log2_trans</span>(),
                            <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">800</span>),
                            <span class="dt">minor_breaks =</span> <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">10</span> <span class="op">*</span><span class="st"> </span><span class="dv">10</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">8</span> <span class="op">*</span><span class="st"> </span><span class="dv">100</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>( <span class="dt">panel.grid.major.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#A9A9A9&quot;</span>,
                                                  <span class="dt">size =</span> <span class="fl">0.6</span>) )
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>( <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#C0C0C0&quot;</span>,
                                                  <span class="dt">size =</span> <span class="fl">0.2</span>) )
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Depth (DP)&quot;</span>)
p</code></pre></div>
<p><img src="quality_control_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We see that the sample P1362 has been removed from the data set, no variants over 100X occur and a minimum coverage of 4X has been used to define reasonable coverage. An important point to make here is that we’re not advocating for anyone to use the same thresholds that we’ve used here. Instead, our goal is to provide you with tools so that you can make the changes you think are appropriate and so that you can visualize the effects of those changes.</p>
</div>
<div id="exercises" class="section level2">
<h2>Exercises</h2>
<p><strong>1)</strong> The <a href="http://samtools.github.io/hts-specs/">VCF specification</a> allows the <code>FORMAT</code> for each variant to be different. How can we learn how many different <code>FORMAT</code>s are in our data?</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-13">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-13" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">unique</span>(vcf<span class="op">@</span>gt[,<span class="dv">1</span>])</code></pre></div>
<pre class="r-output"><code>## [1] "GT:AD:DP:GQ:PL"
</code></pre>
</div>
<p><br /></p>
<p><strong>2)</strong> Our data includes <code>GQ</code>. How can we find out what this means?</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-14">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-14" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">queryMETA</span>(vcf, <span class="dt">element =</span> <span class="st">&quot;=GQ&quot;</span>)</code></pre></div>
<pre class="r-output"><code>## [[1]]
## [1] "FORMAT=ID=GQ"                 "Number=1"                     "Type=Integer"                 "Description=Genotype Quality"
</code></pre>
</div>
<p><br /></p>
<p><strong>3)</strong> How can we extract the <code>GQ</code> information.</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-15">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-15" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gq &lt;-<span class="st"> </span><span class="kw">extract.gt</span>(vcf, <span class="dt">element =</span> <span class="st">&quot;GQ&quot;</span>, <span class="dt">as.numeric =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<p><br /></p>
<p><strong>4)</strong> How can we visualize the <code>GQ</code> data?</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-16">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-16" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dpf &lt;-<span class="st"> </span><span class="kw">melt</span>(gq, <span class="dt">varnames =</span> <span class="kw">c</span>(<span class="st">&quot;Index&quot;</span>, <span class="st">&quot;Sample&quot;</span>),
            <span class="dt">value.name =</span> <span class="st">&quot;GQ&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dpf, <span class="kw">aes</span>(<span class="dt">x =</span> Sample, <span class="dt">y =</span> GQ))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_violin</span>(<span class="dt">fill =</span> <span class="st">&quot;#B22222&quot;</span>, <span class="dt">adjust =</span> <span class="fl">1.0</span>,
                     <span class="dt">scale =</span> <span class="st">&quot;count&quot;</span>, <span class="dt">trim =</span> <span class="ot">TRUE</span>)
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),
               <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>,
                                          <span class="dt">hjust =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">12</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.major.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#A9A9A9&quot;</span>,
                                                 <span class="dt">size =</span> <span class="fl">0.6</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.minor.y =</span> <span class="kw">element_line</span>(<span class="dt">color =</span> <span class="st">&quot;#C0C0C0&quot;</span>,
                                                 <span class="dt">size =</span> <span class="fl">0.2</span>))
p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Genotype quality (GQ)&quot;</span>)
p</code></pre></div>
<img src="quality_control_files/figure-html/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" />
</div>
<p><br /></p>
<p><strong>5)</strong> For the sequence depth data we omitted values that were equal to zero but we have not here. Why was this step necessary for the depth data but not the genotype quality?</p>
<button class="btn btn-danger" data-toggle="collapse" data-target="#hide_buttonunnamed-chunk-17">
Show solution
</button>
<div id="hide_buttonunnamed-chunk-17" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We log transformed the depth data.</span>
<span class="co"># The log of zero is infinite.</span>
<span class="kw">log</span>(<span class="dv">0</span>)</code></pre></div>
<pre class="r-output"><code>## [1] -Inf
</code></pre>
</div>
<p><br /></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
