<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Preparation for the Workshop</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-107144798-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-107144798-3');
</script>


<style type="text/css">code{white-space: pre;}</style>
<<<<<<< HEAD
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
=======
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
>>>>>>> f22ebcda00ffc3afca24a95013bcc0dfb48b93bf
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
<<<<<<< HEAD
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
=======
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
>>>>>>> f22ebcda00ffc3afca24a95013bcc0dfb48b93bf
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
<<<<<<< HEAD
=======

>>>>>>> f22ebcda00ffc3afca24a95013bcc0dfb48b93bf
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and genomics in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="TOC.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part I
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="Getting_ready_to_use_R.html">Getting ready to use R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part II
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Data_Preparation.html">Data preparation</a>
    </li>
    <li>
      <a href="First_Steps.html">First steps</a>
    </li>
    <li>
      <a href="Population_Strata.html">Population strata and clone correction</a>
    </li>
    <li>
      <a href="Locus_Stats.html">Locus-based statistics and missing data</a>
    </li>
    <li>
      <a href="Genotypic_EvenRichDiv.html">Genotypic evenness, richness, and diversity</a>
    </li>
    <li>
      <a href="Linkage_disequilibrium.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="Pop_Structure.html">Population structure</a>
    </li>
    <li>
      <a href="Minimum_Spanning_Networks.html">Minimum Spanning Networks</a>
    </li>
    <li>
      <a href="AMOVA.html">AMOVA</a>
    </li>
    <li>
      <a href="DAPC.html">Discriminant analysis of principal components (DAPC)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Part III
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_vcf.html">Population genomics and HTS</a>
    </li>
    <li>
      <a href="reading_vcf.html">Reading VCF data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genomic data</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="clustering_plot.html">Clustering plot</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Workshop
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="workshop.html">Workshop</a>
    </li>
    <li>
      <a href="intro_vcf.html">Introduction</a>
    </li>
    <li>
      <a href="reading_vcf.html">VCF data</a>
    </li>
    <li>
      <a href="quality_control.html">Quality control</a>
    </li>
    <li>
      <a href="gbs_analysis.html">Analysis of GBS data</a>
    </li>
    <li>
      <a href="analysis_of_genome.html">Analysis of genome data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Authors.html">Authors</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Appendices
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_to_R.html">Introduction to R</a>
    </li>
    <li>
      <a href="Data_sets.html">Data sets</a>
    </li>
    <li>
      <a href="funpendix.html">Function glossary</a>
    </li>
    <li>
      <a href="background_functions.html">Background_functions</a>
    </li>
    <li>
      <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/">Source Code</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preparation for the Workshop</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#population-genomics-workshop">Population genomics workshop</a></li>
<li><a href="#before-you-arrive-please-make-sure-you-install-the-following-required-materials">Before you arrive please make sure you install the following required materials</a></li>
<li><a href="#your-first-homework">Your first homework</a></li>
<li><a href="#workshop-agenda">Workshop agenda</a></li>
<li><a href="#howd-we-do">How’d we do?</a></li>
</ul>
</div>

<div id="population-genomics-workshop" class="section level2">
<h2>Population genomics workshop</h2>
<div id="icpp-boston-massachusetts-2018" class="section level3">
<h3>ICPP, Boston, Massachusetts, 2018</h3>
<p>Previous versions of this material were also presented at:<br />
* APS, San Antonio, Texas, 2017</p>
<p>For workshops, we subset the information available elsewhere in this book to the pages in this section specifically for the workshop. This provides a convenient way to reduce the contents of a book to the essential portions we feel can be reasonably covered in a several hour long workshop. However, the cost of brevity is a lack of detail. While this section may be used as an entry point, learning about genomic analyses in R, details may be found elsewhere in the book.</p>
</div>
</div>
<div id="before-you-arrive-please-make-sure-you-install-the-following-required-materials" class="section level2">
<h2>Before you arrive please make sure you install the following required materials</h2>
<ol style="list-style-type: decimal">
<li><strong>Install R</strong> Instructions on how to install R can be found at the <a href="https://www.r-project.org/">R homepage</a>. Please make sure you have the most current version of R (at least 3.4.4, Someone to lean on).</li>
<li><strong>Install RStudio</strong><br />
Instructions on how to install the RStudio integrated development environment (IDE) can be found at the <a href="https://www.rstudio.com/products/rstudio/download/">RStudio download site</a>. Choose the ‘RStudio Desktop Open Source License’ (free) version.</li>
<li><strong>Install R packages</strong><br />
You can install the R packages we will be using during the workshop by starting R and copying and pasting the below command into the R console.</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">&quot;adegenet&quot;</span>, <span class="st">&quot;ape&quot;</span>, <span class="st">&quot;cowplot&quot;</span>, <span class="st">&quot;devtools&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;hierfstat&quot;</span>, <span class="st">&quot;igraph&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;lattice&quot;</span>, <span class="st">&quot;magrittr&quot;</span>, <span class="st">&quot;mmod&quot;</span>, <span class="st">&quot;pegas&quot;</span>, <span class="st">&quot;pinfsc50&quot;</span>, <span class="st">&quot;poppr&quot;</span>, <span class="st">&quot;RColorBrewer&quot;</span>, <span class="st">&quot;reshape2&quot;</span>, <span class="st">&quot;treemap&quot;</span>, <span class="st">&quot;vcfR&quot;</span>, <span class="st">&quot;vegan&quot;</span>, <span class="st">&quot;viridisLite&quot;</span>))</a></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li><p><strong>Create an RStudio “Project”</strong><br />
Open RStudio and under the “File” dropdown menu select “New Project…”. A “New Project” window should pop up where you should select the option to create your project in a “New Directory”. Select the “New Project” option. You will be asked where on your filesystem you’d like your project and what to name your project. We’ll name our project “APS_Workshop”. This should create a directory for your project that will contain a file named “APS_Workshop.Rproj”. When you double click this file it should spawn a new session of RStudio and will use that directory as your <a href="https://en.wikipedia.org/wiki/Working_directory">working directory</a>.</p></li>
<li><p><strong>Download the example datasets </strong><br />
We have placed example data files at the OSF site for <a href="https://osf.io/ytd6k/">Population genomics in R workshop</a>. You can download them by copying and pasting the below code into your R console.</p></li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">download.file</span>(<span class="st">&quot;https://osf.io/td5sx/download&quot;</span>, <span class="dt">destfile =</span> <span class="st">&quot;pinfsc50_filtered.vcf.gz&quot;</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">download.file</span>(<span class="st">&quot;https://osf.io/kaq5p/download&quot;</span>, <span class="dt">destfile =</span> <span class="st">&quot;population_data.gbs.txt&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">download.file</span>(<span class="st">&quot;https://osf.io/c6h4t/download&quot;</span>, <span class="dt">destfile =</span> <span class="st">&quot;prubi_gbs.vcf.gz&quot;</span>)</a></code></pre></div>
<p>You should be able to validate that the files have been downloaded with the below code.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">list.files</span>(<span class="dt">pattern =</span> <span class="st">&quot;.vcf.gz$|.gbs.txt&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;pinfsc50_filtered.vcf.gz&quot; &quot;population_data.gbs.txt&quot; 
## [3] &quot;prubi_gbs.vcf.gz&quot;</code></pre>
<ol start="6" style="list-style-type: decimal">
<li><strong>Test that your system is ready:</strong><br />
You should now be able to validate that the resources you need are ready. Change directory to <code>APS_Workshop</code> and copy and paste the below command into the R console.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="cf">if</span>(<span class="kw">require</span>(devtools)){</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">  devtools<span class="op">::</span><span class="kw">source_gist</span>(<span class="st">&quot;01a3d8efb21e0a4dbac0735270d147af&quot;</span>, <span class="dt">filename=</span><span class="st">&quot;apstest.R&quot;</span>)       </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  <span class="kw">print</span>(<span class="st">&quot;Please install the package devtools.&quot;</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">  <span class="kw">print</span>(<span class="st">&quot;Use: install.packages(&#39;devtools&#39;)&quot;</span>)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">}</a></code></pre></div>
<p>This should output some tests to the console. It should also generate a report file called <code>apstest.txt</code>. The report file should look exactly like <a href="https://github.com/grunwaldlab/Population_Genetics_in_R/blob/master/apstest.txt">this file</a>. If your results are different, and you don’t understand why, send us an email with the report as an attachment.</p>
<ol start="7" style="list-style-type: decimal">
<li><strong>If you have never worked with R, please review the <a href="intro_to_R.html">Introduction to R</a> page</strong></li>
</ol>
</div>
<div id="your-first-homework" class="section level2">
<h2>Your first homework</h2>
<p>Some aspects of analyzing genetic data are rather technical. Others are more stylistic. An example is the presentation of the data. The presentation of data may include choices in color schemes and sometimes the perspective on the data. Below is a little example that you can copy and paste into your R console. Explore how changing the number in <code>set.seed()</code> changes the plot. Remember to execute the <code>plot_poppr_msn(partial_clone, myMsn, palette = brewer.pal(n=4, name = &quot;Set1&quot;))</code> function again. There are other examples that are commented out (i.e., the lines begin with <code>#</code> so they are not executed). Try removing the comment character (<code>#</code>) and see how the different parameters affect the plot. This example should validate that you have successfully installed <code>poppr</code> and hopefully provides a fun example that may inspire you to explore more options.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">library</span>(viridisLite)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">library</span>(poppr)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw">data</span>(partial_clone)</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">myMsn &lt;-<span class="st"> </span><span class="kw">bruvo.msn</span>(partial_clone, <span class="dt">include.ties =</span> <span class="ot">TRUE</span>, <span class="dt">showplot =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="kw">set.seed</span>(<span class="dv">9</span>)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="kw">plot_poppr_msn</span>(partial_clone, myMsn, <span class="dt">palette =</span> <span class="kw">brewer.pal</span>(<span class="dt">n=</span><span class="dv">4</span>, <span class="dt">name =</span> <span class="st">&quot;Set1&quot;</span>))</a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#plot_poppr_msn(partial_clone, myMsn, palette = magma(n=4, begin = 0.2, end = 0.8))</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#plot_poppr_msn(partial_clone, myMsn, palette = plasma(n=4))</span></a></code></pre></div>
</div>
<div id="workshop-agenda" class="section level2">
<h2>Workshop agenda</h2>
<ol style="list-style-type: decimal">
<li>Introduction<br />
</li>
<li>Introduction to <a href="reading_vcf.html">VCF data</a><br />
</li>
<li>Analysis of <a href="gbs_analysis.html">Genotyping by sequencing</a> data</li>
<li>Analysis of <a href="analysis_of_genome.html">Genome</a> data</li>
</ol>
<hr />
</div>
<div id="howd-we-do" class="section level2">
<h2>How’d we do?</h2>
<p>After you’ve completed the workshop, please fill out <a href="https://docs.google.com/forms/d/1Buc6fsnRxEtk0QhMxH_b3yPFhP3du83zb-4m7Hd9Xtc/edit">this quick survey</a> to provide us with feedback. Thank you!</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
